{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import joblib\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "dict_keys(['text', 'segments', 'language'])\n",
      "[{'id': 0, 'seek': 0, 'start': 0.0, 'end': 1.0, 'text': ' Thank you all.', 'tokens': [50363, 6952, 345, 477, 13, 50413], 'temperature': 0.0, 'avg_logprob': -0.45480142982260696, 'compression_ratio': 1.611353711790393, 'no_speech_prob': 0.2748943865299225}, {'id': 1, 'seek': 0, 'start': 1.0, 'end': 2.0, 'text': ' Good afternoon.', 'tokens': [50413, 4599, 6672, 13, 50463], 'temperature': 0.0, 'avg_logprob': -0.45480142982260696, 'compression_ratio': 1.611353711790393, 'no_speech_prob': 0.2748943865299225}, {'id': 2, 'seek': 0, 'start': 2.0, 'end': 3.0, 'text': ' How are you guys doing?', 'tokens': [50463, 1374, 389, 345, 3730, 1804, 30, 50513], 'temperature': 0.0, 'avg_logprob': -0.45480142982260696, 'compression_ratio': 1.611353711790393, 'no_speech_prob': 0.2748943865299225}, {'id': 3, 'seek': 0, 'start': 3.0, 'end': 4.0, 'text': ' It looks fun.', 'tokens': [50513, 632, 3073, 1257, 13, 50563], 'temperature': 0.0, 'avg_logprob': -0.45480142982260696, 'compression_ratio': 1.611353711790393, 'no_speech_prob': 0.2748943865299225}, {'id': 4, 'seek': 0, 'start': 4.0, 'end': 7.0, 'text': ' How many fans have done themselves last week?', 'tokens': [50563, 1374, 867, 3296, 423, 1760, 2405, 938, 1285, 30, 50713], 'temperature': 0.0, 'avg_logprob': -0.45480142982260696, 'compression_ratio': 1.611353711790393, 'no_speech_prob': 0.2748943865299225}, {'id': 5, 'seek': 0, 'start': 7.0, 'end': 8.0, 'text': ' Too easy?', 'tokens': [50713, 14190, 2562, 30, 50763], 'temperature': 0.0, 'avg_logprob': -0.45480142982260696, 'compression_ratio': 1.611353711790393, 'no_speech_prob': 0.2748943865299225}, {'id': 6, 'seek': 0, 'start': 8.0, 'end': 16.0, 'text': ' At least, maybe you think, well, applied the concepts that we learned in the past.', 'tokens': [50763, 1629, 1551, 11, 3863, 345, 892, 11, 880, 11, 5625, 262, 10838, 326, 356, 4499, 287, 262, 1613, 13, 51163], 'temperature': 0.0, 'avg_logprob': -0.45480142982260696, 'compression_ratio': 1.611353711790393, 'no_speech_prob': 0.2748943865299225}, {'id': 7, 'seek': 0, 'start': 16.0, 'end': 21.0, 'text': ' I hope that we all have relevant problems, for modern students like you guys.', 'tokens': [51163, 314, 2911, 326, 356, 477, 423, 5981, 2761, 11, 329, 3660, 2444, 588, 345, 3730, 13, 51413], 'temperature': 0.0, 'avg_logprob': -0.45480142982260696, 'compression_ratio': 1.611353711790393, 'no_speech_prob': 0.2748943865299225}, {'id': 8, 'seek': 0, 'start': 21.0, 'end': 27.0, 'text': ' At least, I hope you had some fun from doing it and not finding it too challenging.', 'tokens': [51413, 1629, 1551, 11, 314, 2911, 345, 550, 617, 1257, 422, 1804, 340, 290, 407, 4917, 340, 1165, 9389, 13, 51713], 'temperature': 0.0, 'avg_logprob': -0.45480142982260696, 'compression_ratio': 1.611353711790393, 'no_speech_prob': 0.2748943865299225}, {'id': 9, 'seek': 2700, 'start': 27.0, 'end': 36.0, 'text': ' The final exam questions, they will be largely in the context of problem solving like this.', 'tokens': [50363, 383, 2457, 2814, 2683, 11, 484, 481, 307, 5688, 287, 262, 4732, 286, 1917, 18120, 588, 428, 13, 50813], 'temperature': 0.0, 'avg_logprob': -0.28589046833127046, 'compression_ratio': 1.7281553398058251, 'no_speech_prob': 0.6804109215736389}, {'id': 10, 'seek': 2700, 'start': 36.0, 'end': 40.0, 'text': ' And the level of challenge will be suddenly a little bit less, perhaps,', 'tokens': [50813, 843, 262, 1241, 286, 4427, 481, 307, 6451, 257, 1310, 1643, 1342, 11, 3737, 11, 51013], 'temperature': 0.0, 'avg_logprob': -0.28589046833127046, 'compression_ratio': 1.7281553398058251, 'no_speech_prob': 0.6804109215736389}, {'id': 11, 'seek': 2700, 'start': 40.0, 'end': 43.0, 'text': ' because in the final exam you are kind of limited on it.', 'tokens': [51013, 780, 287, 262, 2457, 2814, 345, 389, 1611, 286, 3614, 319, 340, 13, 51163], 'temperature': 0.0, 'avg_logprob': -0.28589046833127046, 'compression_ratio': 1.7281553398058251, 'no_speech_prob': 0.6804109215736389}, {'id': 12, 'seek': 2700, 'start': 43.0, 'end': 47.0, 'text': ' So the highest level of challenge you will see, and at least my point in the course,', 'tokens': [51163, 1406, 262, 4511, 1241, 286, 4427, 345, 481, 766, 11, 290, 379, 1551, 616, 966, 287, 262, 1781, 11, 51363], 'temperature': 0.0, 'avg_logprob': -0.28589046833127046, 'compression_ratio': 1.7281553398058251, 'no_speech_prob': 0.6804109215736389}, {'id': 13, 'seek': 2700, 'start': 47.0, 'end': 53.0, 'text': ' will be represented by the problem sets questions.', 'tokens': [51363, 481, 307, 7997, 416, 262, 1917, 5621, 2683, 13, 51663], 'temperature': 0.0, 'avg_logprob': -0.28589046833127046, 'compression_ratio': 1.7281553398058251, 'no_speech_prob': 0.6804109215736389}, {'id': 14, 'seek': 5300, 'start': 53.0, 'end': 59.0, 'text': ' So if you feel that we understand them and you can tackle them, especially after the session,', 'tokens': [50363, 1406, 611, 345, 1254, 326, 356, 1833, 606, 290, 345, 460, 9761, 606, 11, 2592, 706, 262, 6246, 11, 50663], 'temperature': 0.0, 'avg_logprob': -0.2227006477884727, 'compression_ratio': 1.592057761732852, 'no_speech_prob': 0.33141064643859863}, {'id': 15, 'seek': 5300, 'start': 59.0, 'end': 64.0, 'text': ' and you are engaged in the class and you really should not need to worry suddenly from the', 'tokens': [50663, 290, 345, 389, 7953, 287, 262, 1398, 290, 345, 1107, 815, 407, 761, 284, 5490, 6451, 422, 262, 50913], 'temperature': 0.0, 'avg_logprob': -0.2227006477884727, 'compression_ratio': 1.592057761732852, 'no_speech_prob': 0.33141064643859863}, {'id': 16, 'seek': 5300, 'start': 64.0, 'end': 69.0, 'text': \" three lecture series that I'm very happy to provide here for this course.\", 'tokens': [50913, 1115, 19143, 2168, 326, 314, 1101, 845, 3772, 284, 2148, 994, 329, 428, 1781, 13, 51163], 'temperature': 0.0, 'avg_logprob': -0.2227006477884727, 'compression_ratio': 1.592057761732852, 'no_speech_prob': 0.33141064643859863}, {'id': 17, 'seek': 5300, 'start': 69.0, 'end': 72.0, 'text': \" So let's continue to today's topic too.\", 'tokens': [51163, 1406, 1309, 338, 2555, 284, 1909, 338, 7243, 1165, 13, 51313], 'temperature': 0.0, 'avg_logprob': -0.2227006477884727, 'compression_ratio': 1.592057761732852, 'no_speech_prob': 0.33141064643859863}, {'id': 18, 'seek': 5300, 'start': 72.0, 'end': 76.0, 'text': \" We're going to learn about designing and applying these RNA switches.\", 'tokens': [51313, 775, 821, 1016, 284, 2193, 546, 18492, 290, 11524, 777, 25897, 18225, 13, 51513], 'temperature': 0.0, 'avg_logprob': -0.2227006477884727, 'compression_ratio': 1.592057761732852, 'no_speech_prob': 0.33141064643859863}, {'id': 19, 'seek': 5300, 'start': 76.0, 'end': 80.0, 'text': ' You probably know synthetic biologists are very much inspired by nature,', 'tokens': [51513, 921, 2192, 760, 18512, 43149, 389, 845, 881, 7867, 416, 3450, 11, 51713], 'temperature': 0.0, 'avg_logprob': -0.2227006477884727, 'compression_ratio': 1.592057761732852, 'no_speech_prob': 0.33141064643859863}, {'id': 20, 'seek': 8000, 'start': 80.0, 'end': 86.0, 'text': ' by how nature designs these circuits and controls cell decision making and cell responses,', 'tokens': [50363, 416, 703, 3450, 9824, 777, 24907, 290, 6973, 2685, 2551, 1642, 290, 2685, 9109, 11, 50663], 'temperature': 0.0, 'avg_logprob': -0.2217906399777061, 'compression_ratio': 1.5270935960591132, 'no_speech_prob': 0.17531506717205048}, {'id': 21, 'seek': 8000, 'start': 86.0, 'end': 88.0, 'text': ' all the way at the animal level.', 'tokens': [50663, 477, 262, 835, 379, 262, 5044, 1241, 13, 50763], 'temperature': 0.0, 'avg_logprob': -0.2217906399777061, 'compression_ratio': 1.5270935960591132, 'no_speech_prob': 0.17531506717205048}, {'id': 22, 'seek': 8000, 'start': 88.0, 'end': 96.0, 'text': ' And when we use these ideas, we really be able to control back cells and living culture models.', 'tokens': [50763, 843, 618, 356, 779, 777, 4213, 11, 356, 1107, 307, 1498, 284, 1630, 736, 4778, 290, 2877, 3968, 4981, 13, 51163], 'temperature': 0.0, 'avg_logprob': -0.2217906399777061, 'compression_ratio': 1.5270935960591132, 'no_speech_prob': 0.17531506717205048}, {'id': 23, 'seek': 8000, 'start': 96.0, 'end': 99.0, 'text': ' So we will focus on RNA level switches.', 'tokens': [51163, 1406, 356, 481, 2962, 319, 25897, 1241, 18225, 13, 51313], 'temperature': 0.0, 'avg_logprob': -0.2217906399777061, 'compression_ratio': 1.5270935960591132, 'no_speech_prob': 0.17531506717205048}, {'id': 24, 'seek': 8000, 'start': 99.0, 'end': 104.0, 'text': ' Last week it was therapeutics, as you guys recall.', 'tokens': [51313, 4586, 1285, 340, 373, 20094, 48063, 11, 355, 345, 3730, 10014, 13, 51563], 'temperature': 0.0, 'avg_logprob': -0.2217906399777061, 'compression_ratio': 1.5270935960591132, 'no_speech_prob': 0.17531506717205048}, {'id': 25, 'seek': 10400, 'start': 105.0, 'end': 111.0, 'text': ' So the key aspects that we will target will be first going through the principles of natural regulatory', 'tokens': [50413, 1406, 262, 1994, 7612, 326, 356, 481, 2496, 481, 307, 717, 1016, 832, 262, 7811, 286, 3288, 11344, 50713], 'temperature': 0.0, 'avg_logprob': -0.13671333010834041, 'compression_ratio': 1.6666666666666667, 'no_speech_prob': 0.0655023604631424}, {'id': 26, 'seek': 10400, 'start': 111.0, 'end': 117.0, 'text': \" RNA. That's kind of important fundamentals because we need to know how nature does these mechanisms,\", 'tokens': [50713, 25897, 13, 1320, 338, 1611, 286, 1593, 33099, 780, 356, 761, 284, 760, 703, 3450, 857, 777, 11701, 11, 51013], 'temperature': 0.0, 'avg_logprob': -0.13671333010834041, 'compression_ratio': 1.6666666666666667, 'no_speech_prob': 0.0655023604631424}, {'id': 27, 'seek': 10400, 'start': 117.0, 'end': 122.0, 'text': ' and how we can hijack these principles to really control the cell decision making.', 'tokens': [51013, 290, 703, 356, 460, 16836, 441, 777, 7811, 284, 1107, 1630, 262, 2685, 2551, 1642, 13, 51263], 'temperature': 0.0, 'avg_logprob': -0.13671333010834041, 'compression_ratio': 1.6666666666666667, 'no_speech_prob': 0.0655023604631424}, {'id': 28, 'seek': 10400, 'start': 122.0, 'end': 127.0, 'text': ' And there will be three courses that we will concern with for our course level.', 'tokens': [51263, 843, 612, 481, 307, 1115, 10902, 326, 356, 481, 2328, 351, 329, 674, 1781, 1241, 13, 51513], 'temperature': 0.0, 'avg_logprob': -0.13671333010834041, 'compression_ratio': 1.6666666666666667, 'no_speech_prob': 0.0655023604631424}, {'id': 29, 'seek': 10400, 'start': 127.0, 'end': 132.0, 'text': ' This will be regulators, ribozymes, RNA-based enzymes, and riboswitches.', 'tokens': [51513, 770, 481, 307, 17199, 11, 12183, 8590, 22009, 11, 25897, 12, 3106, 29120, 11, 290, 12183, 418, 86, 9249, 13, 51763], 'temperature': 0.0, 'avg_logprob': -0.13671333010834041, 'compression_ratio': 1.6666666666666667, 'no_speech_prob': 0.0655023604631424}, {'id': 30, 'seek': 13200, 'start': 132.0, 'end': 139.0, 'text': ' And lastly, connected to the second topic, it will be small molecule ligand or drug-guided RNA conformational gating', 'tokens': [50363, 843, 938, 306, 11, 5884, 284, 262, 1218, 7243, 11, 340, 481, 307, 1402, 27756, 26106, 392, 393, 2563, 12, 23657, 25897, 17216, 864, 308, 803, 50713], 'temperature': 0.0, 'avg_logprob': -0.13429405319858606, 'compression_ratio': 1.519047619047619, 'no_speech_prob': 0.006292225327342749}, {'id': 31, 'seek': 13200, 'start': 139.0, 'end': 146.0, 'text': ' that will allow you to achieve this reversible regulatory control at the level of these RNA switches.', 'tokens': [50713, 326, 481, 1249, 345, 284, 4620, 428, 48287, 11344, 1630, 379, 262, 1241, 286, 777, 25897, 18225, 13, 51063], 'temperature': 0.0, 'avg_logprob': -0.13429405319858606, 'compression_ratio': 1.519047619047619, 'no_speech_prob': 0.006292225327342749}, {'id': 32, 'seek': 13200, 'start': 148.0, 'end': 156.0, 'text': \" And first, to cover everyone's kind of basic fundamental principles on ribosome protein translation.\", 'tokens': [51163, 843, 717, 11, 284, 3002, 2506, 338, 1611, 286, 4096, 7531, 7811, 319, 12183, 418, 462, 7532, 11059, 13, 51563], 'temperature': 0.0, 'avg_logprob': -0.13429405319858606, 'compression_ratio': 1.519047619047619, 'no_speech_prob': 0.006292225327342749}, {'id': 33, 'seek': 15600, 'start': 156.0, 'end': 163.0, 'text': ' Pretty much last week, we talked about this central dogma, right, starting from transcription and translation,', 'tokens': [50363, 20090, 881, 938, 1285, 11, 356, 6619, 546, 428, 4318, 40937, 11, 826, 11, 3599, 422, 26955, 290, 11059, 11, 50713], 'temperature': 0.0, 'avg_logprob': -0.16720433351470204, 'compression_ratio': 1.6724137931034482, 'no_speech_prob': 0.05509773641824722}, {'id': 34, 'seek': 15600, 'start': 163.0, 'end': 168.0, 'text': \" and of course there's also this reverse transcription method that's happened in nature in certain contexts.\", 'tokens': [50713, 290, 286, 1781, 612, 338, 635, 428, 9575, 26955, 2446, 326, 338, 3022, 287, 3450, 287, 1728, 26307, 13, 50963], 'temperature': 0.0, 'avg_logprob': -0.16720433351470204, 'compression_ratio': 1.6724137931034482, 'no_speech_prob': 0.05509773641824722}, {'id': 35, 'seek': 15600, 'start': 168.0, 'end': 174.0, 'text': ' But what about this second step to the I mentioned here, protein translation.', 'tokens': [50963, 887, 644, 546, 428, 1218, 2239, 284, 262, 314, 4750, 994, 11, 7532, 11059, 13, 51263], 'temperature': 0.0, 'avg_logprob': -0.16720433351470204, 'compression_ratio': 1.6724137931034482, 'no_speech_prob': 0.05509773641824722}, {'id': 36, 'seek': 15600, 'start': 174.0, 'end': 183.0, 'text': ' You guys know what machinery in any organism that does this translation, what is it called?', 'tokens': [51263, 921, 3730, 760, 644, 20230, 287, 597, 26433, 326, 857, 428, 11059, 11, 644, 318, 340, 1444, 30, 51713], 'temperature': 0.0, 'avg_logprob': -0.16720433351470204, 'compression_ratio': 1.6724137931034482, 'no_speech_prob': 0.05509773641824722}, {'id': 37, 'seek': 18300, 'start': 183.0, 'end': 190.0, 'text': \" Machinery, biomolecular machine that does this translation. It's actually known as ribosome.\", 'tokens': [50363, 7080, 15451, 11, 27488, 2305, 10440, 4572, 326, 857, 428, 11059, 13, 632, 338, 1682, 1900, 355, 12183, 418, 462, 13, 50713], 'temperature': 0.0, 'avg_logprob': -0.1364655918545193, 'compression_ratio': 1.6355140186915889, 'no_speech_prob': 0.012160781770944595}, {'id': 38, 'seek': 18300, 'start': 190.0, 'end': 196.0, 'text': \" So we'll talk a little bit about ribosome. So as you can see in the footnote, it's a protein synthesis machine in the cell.\", 'tokens': [50713, 1406, 356, 1183, 1561, 257, 1310, 1643, 546, 12183, 418, 462, 13, 1406, 355, 345, 460, 766, 287, 262, 38512, 11, 340, 338, 257, 7532, 21263, 4572, 287, 262, 2685, 13, 51013], 'temperature': 0.0, 'avg_logprob': -0.1364655918545193, 'compression_ratio': 1.6355140186915889, 'no_speech_prob': 0.012160781770944595}, {'id': 39, 'seek': 18300, 'start': 196.0, 'end': 207.0, 'text': ' It is a molecular, macromolecular complex made up of protein subunits together with certain RNAs that you will see is a catalytic RNA', 'tokens': [51013, 632, 318, 257, 18955, 11, 8352, 398, 2305, 10440, 3716, 925, 510, 286, 7532, 850, 41667, 1978, 351, 1728, 43112, 1722, 326, 345, 481, 766, 318, 257, 36745, 13370, 25897, 51563], 'temperature': 0.0, 'avg_logprob': -0.1364655918545193, 'compression_ratio': 1.6355140186915889, 'no_speech_prob': 0.012160781770944595}, {'id': 40, 'seek': 20700, 'start': 207.0, 'end': 216.0, 'text': ' that essentially together, this machine allows you to convert the mRNA sequence to your protein sequence.', 'tokens': [50363, 326, 6986, 1978, 11, 428, 4572, 3578, 345, 284, 10385, 262, 47227, 8379, 284, 534, 7532, 8379, 13, 50813], 'temperature': 0.0, 'avg_logprob': -0.16713451385498046, 'compression_ratio': 1.6323529411764706, 'no_speech_prob': 0.3385063707828522}, {'id': 41, 'seek': 20700, 'start': 216.0, 'end': 225.0, 'text': ' Proteins are made up of amino acids that essentially is a long polymer of individual 20-coding amino acids.', 'tokens': [50813, 1041, 660, 1040, 389, 925, 510, 286, 23206, 17045, 326, 6986, 318, 257, 890, 30528, 286, 1981, 1160, 12, 66, 7656, 23206, 17045, 13, 51263], 'temperature': 0.0, 'avg_logprob': -0.16713451385498046, 'compression_ratio': 1.6323529411764706, 'no_speech_prob': 0.3385063707828522}, {'id': 42, 'seek': 20700, 'start': 225.0, 'end': 233.0, 'text': \" So you can imagine this machine, it doesn't have in all cells, including bacteria and bacteria, those bacteria like us,\", 'tokens': [51263, 1406, 345, 460, 5967, 428, 4572, 11, 340, 1595, 470, 423, 287, 477, 4778, 11, 1390, 11492, 290, 11492, 11, 883, 11492, 588, 514, 11, 51663], 'temperature': 0.0, 'avg_logprob': -0.16713451385498046, 'compression_ratio': 1.6323529411764706, 'no_speech_prob': 0.3385063707828522}, {'id': 43, 'seek': 23300, 'start': 233.0, 'end': 243.0, 'text': \" essentially it's a polymerization machine if you think about it, because you're going to allow amino acids, individual amino acids to be covalently linked.\", 'tokens': [50363, 6986, 340, 338, 257, 30528, 1634, 4572, 611, 345, 892, 546, 340, 11, 780, 345, 821, 1016, 284, 1249, 23206, 17045, 11, 1981, 23206, 17045, 284, 307, 269, 8325, 1473, 6692, 13, 50863], 'temperature': 0.0, 'avg_logprob': -0.20949992679414295, 'compression_ratio': 1.60352422907489, 'no_speech_prob': 0.07042592763900757}, {'id': 44, 'seek': 23300, 'start': 243.0, 'end': 256.0, 'text': ' In peptide bonds, so-called A-migons, you have these long chain proteins that you know that proteins are absolutely essential macromolecules that actually drive the cellular engine, just like in a car engine.', 'tokens': [50863, 554, 34337, 485, 13100, 11, 523, 12, 7174, 317, 12, 76, 328, 684, 11, 345, 423, 777, 890, 6333, 15568, 326, 345, 760, 326, 15568, 389, 5543, 6393, 8352, 398, 2305, 13930, 326, 1682, 3708, 262, 19824, 3113, 11, 655, 588, 287, 257, 1097, 3113, 13, 51513], 'temperature': 0.0, 'avg_logprob': -0.20949992679414295, 'compression_ratio': 1.60352422907489, 'no_speech_prob': 0.07042592763900757}, {'id': 45, 'seek': 25600, 'start': 256.0, 'end': 263.0, 'text': ' So that is a very essential aspect of protein synthesis is a ribosome.', 'tokens': [50363, 1406, 326, 318, 257, 845, 6393, 4843, 286, 7532, 21263, 318, 257, 12183, 418, 462, 13, 50713], 'temperature': 0.0, 'avg_logprob': -0.16153527469169804, 'compression_ratio': 1.616580310880829, 'no_speech_prob': 0.34936168789863586}, {'id': 46, 'seek': 25600, 'start': 263.0, 'end': 270.0, 'text': ' So before I go there, just give you an idea of, you know, you may wonder, where does protein synthesis happen?', 'tokens': [50713, 1406, 878, 314, 467, 612, 11, 655, 1577, 345, 281, 2126, 286, 11, 345, 760, 11, 345, 743, 4240, 11, 810, 857, 7532, 21263, 1645, 30, 51063], 'temperature': 0.0, 'avg_logprob': -0.16153527469169804, 'compression_ratio': 1.616580310880829, 'no_speech_prob': 0.34936168789863586}, {'id': 47, 'seek': 25600, 'start': 270.0, 'end': 274.0, 'text': \" What I'm showing you here is the architecture of a eukaryotic cell.\", 'tokens': [51063, 1867, 314, 1101, 4478, 345, 994, 318, 262, 10959, 286, 257, 304, 2724, 560, 6210, 2685, 13, 51263], 'temperature': 0.0, 'avg_logprob': -0.16153527469169804, 'compression_ratio': 1.616580310880829, 'no_speech_prob': 0.34936168789863586}, {'id': 48, 'seek': 25600, 'start': 274.0, 'end': 277.0, 'text': ' Why do you think protein synthesis happens?', 'tokens': [51263, 4162, 466, 345, 892, 7532, 21263, 4325, 30, 51413], 'temperature': 0.0, 'avg_logprob': -0.16153527469169804, 'compression_ratio': 1.616580310880829, 'no_speech_prob': 0.34936168789863586}, {'id': 49, 'seek': 25600, 'start': 277.0, 'end': 284.0, 'text': ' Which compartment?', 'tokens': [51413, 9022, 26247, 30, 51763], 'temperature': 0.0, 'avg_logprob': -0.16153527469169804, 'compression_ratio': 1.616580310880829, 'no_speech_prob': 0.34936168789863586}, {'id': 50, 'seek': 28400, 'start': 284.0, 'end': 292.0, 'text': ' In a very specific location, the majority is on the ER surface, but largely speaking, does it happen in the nucleus of the cytoplasm?', 'tokens': [50363, 554, 257, 845, 2176, 4067, 11, 262, 3741, 318, 319, 262, 13793, 4417, 11, 475, 5688, 5486, 11, 857, 340, 1645, 287, 262, 29984, 286, 262, 3075, 4852, 75, 8597, 30, 50763], 'temperature': 0.0, 'avg_logprob': -0.1610228157043457, 'compression_ratio': 1.7649769585253456, 'no_speech_prob': 0.410399466753006}, {'id': 51, 'seek': 28400, 'start': 292.0, 'end': 293.0, 'text': ' Which compartment?', 'tokens': [50763, 9022, 26247, 30, 50813], 'temperature': 0.0, 'avg_logprob': -0.1610228157043457, 'compression_ratio': 1.7649769585253456, 'no_speech_prob': 0.410399466753006}, {'id': 52, 'seek': 28400, 'start': 293.0, 'end': 295.0, 'text': ' Cytoplasm, right?', 'tokens': [50813, 5934, 4852, 75, 8597, 11, 826, 30, 50913], 'temperature': 0.0, 'avg_logprob': -0.1610228157043457, 'compression_ratio': 1.7649769585253456, 'no_speech_prob': 0.410399466753006}, {'id': 53, 'seek': 28400, 'start': 295.0, 'end': 299.0, 'text': ' Whereas mRNA synthesis, where does it happen in this eukaryotic cell?', 'tokens': [50913, 28016, 47227, 21263, 11, 810, 857, 340, 1645, 287, 428, 304, 2724, 560, 6210, 2685, 30, 51113], 'temperature': 0.0, 'avg_logprob': -0.1610228157043457, 'compression_ratio': 1.7649769585253456, 'no_speech_prob': 0.410399466753006}, {'id': 54, 'seek': 28400, 'start': 299.0, 'end': 302.0, 'text': ' In the nucleus, right?', 'tokens': [51113, 554, 262, 29984, 11, 826, 30, 51263], 'temperature': 0.0, 'avg_logprob': -0.1610228157043457, 'compression_ratio': 1.7649769585253456, 'no_speech_prob': 0.410399466753006}, {'id': 55, 'seek': 28400, 'start': 302.0, 'end': 309.0, 'text': ' So there is this really compartmentalized segregation in eukaryotic system where mRNA synthesis happens in the nucleus,', 'tokens': [51263, 1406, 612, 318, 428, 1107, 26247, 282, 1143, 26718, 287, 304, 2724, 560, 6210, 1080, 810, 47227, 21263, 4325, 287, 262, 29984, 11, 51613], 'temperature': 0.0, 'avg_logprob': -0.1610228157043457, 'compression_ratio': 1.7649769585253456, 'no_speech_prob': 0.410399466753006}, {'id': 56, 'seek': 30900, 'start': 309.0, 'end': 318.0, 'text': ' and this mRNA undergoes maturation processing, ultimately get exported outside the nucleus into the cytoplasm,', 'tokens': [50363, 290, 428, 47227, 17777, 274, 285, 36921, 7587, 11, 6165, 651, 29050, 2354, 262, 29984, 656, 262, 3075, 4852, 75, 8597, 11, 50813], 'temperature': 0.0, 'avg_logprob': -0.1150209880111241, 'compression_ratio': 1.619047619047619, 'no_speech_prob': 0.23736900091171265}, {'id': 57, 'seek': 30900, 'start': 318.0, 'end': 328.0, 'text': ' where it will meet this machinery that I mentioned, ribosome, that will catalyze the polypeptide, i.e. amino acids polymerization, to give you the protein.', 'tokens': [50813, 810, 340, 481, 1826, 428, 20230, 326, 314, 4750, 11, 12183, 418, 462, 11, 326, 481, 36745, 2736, 262, 7514, 431, 457, 485, 11, 1312, 13, 68, 13, 23206, 17045, 30528, 1634, 11, 284, 1577, 345, 262, 7532, 13, 51313], 'temperature': 0.0, 'avg_logprob': -0.1150209880111241, 'compression_ratio': 1.619047619047619, 'no_speech_prob': 0.23736900091171265}, {'id': 58, 'seek': 30900, 'start': 328.0, 'end': 337.0, 'text': ' So these individual black dots, you may see better in your slides, these are actually ribosome, these sort of a complex of RNAs and proteins.', 'tokens': [51313, 1406, 777, 1981, 2042, 22969, 11, 345, 743, 766, 1365, 287, 534, 19392, 11, 777, 389, 1682, 12183, 418, 462, 11, 777, 3297, 286, 257, 3716, 286, 43112, 1722, 290, 15568, 13, 51763], 'temperature': 0.0, 'avg_logprob': -0.1150209880111241, 'compression_ratio': 1.619047619047619, 'no_speech_prob': 0.23736900091171265}, {'id': 59, 'seek': 33700, 'start': 337.0, 'end': 345.0, 'text': ' They are essentially type complex. They are individual ribosome complex molecules that are diffusing throughout the cytoplasm.', 'tokens': [50363, 1119, 389, 6986, 2099, 3716, 13, 1119, 389, 1981, 12183, 418, 462, 3716, 17745, 326, 389, 814, 3500, 3690, 262, 3075, 4852, 75, 8597, 13, 50763], 'temperature': 0.0, 'avg_logprob': -0.12742719532530986, 'compression_ratio': 1.580188679245283, 'no_speech_prob': 0.06587988883256912}, {'id': 60, 'seek': 33700, 'start': 345.0, 'end': 358.0, 'text': \" So protein synthesis actually happens in the cytoplasm. It is true, the student over there said, it's also concentrated on the ER, so these tubules, tubular green structures you see are endoplasmic reticulum.\", 'tokens': [50763, 1406, 7532, 21263, 1682, 4325, 287, 262, 3075, 4852, 75, 8597, 13, 632, 318, 2081, 11, 262, 3710, 625, 612, 531, 11, 340, 338, 635, 17298, 319, 262, 13793, 11, 523, 777, 12202, 5028, 11, 12202, 934, 4077, 8573, 345, 766, 389, 886, 20106, 8597, 291, 1005, 291, 14452, 13, 51413], 'temperature': 0.0, 'avg_logprob': -0.12742719532530986, 'compression_ratio': 1.580188679245283, 'no_speech_prob': 0.06587988883256912}, {'id': 61, 'seek': 35800, 'start': 358.0, 'end': 372.0, 'text': \" So we won't worry so much in our cost level, but essentially the key point is that protein synthesis happens outside the nucleus, broadly speaking in the cytoplasm, which is very different from bacteria system like E. coli.\", 'tokens': [50363, 1406, 356, 1839, 470, 5490, 523, 881, 287, 674, 1575, 1241, 11, 475, 6986, 262, 1994, 966, 318, 326, 7532, 21263, 4325, 2354, 262, 29984, 11, 18633, 5486, 287, 262, 3075, 4852, 75, 8597, 11, 543, 318, 845, 1180, 422, 11492, 1080, 588, 412, 13, 40395, 13, 51063], 'temperature': 0.0, 'avg_logprob': -0.1492058535416921, 'compression_ratio': 1.630952380952381, 'no_speech_prob': 0.6753475666046143}, {'id': 62, 'seek': 35800, 'start': 372.0, 'end': 385.0, 'text': \" In E. coli, do we worry about cytoplasm versus nucleus? The answer is no, right? There's essentially translation and transcription couple process in E. coli, as we talked about last week.\", 'tokens': [51063, 554, 412, 13, 40395, 11, 466, 356, 5490, 546, 3075, 4852, 75, 8597, 9051, 29984, 30, 383, 3280, 318, 645, 11, 826, 30, 1318, 338, 6986, 11059, 290, 26955, 3155, 1429, 287, 412, 13, 40395, 11, 355, 356, 6619, 546, 938, 1285, 13, 51713], 'temperature': 0.0, 'avg_logprob': -0.1492058535416921, 'compression_ratio': 1.630952380952381, 'no_speech_prob': 0.6753475666046143}, {'id': 63, 'seek': 38500, 'start': 385.0, 'end': 400.0, 'text': ' So you may wonder, right, how fast are these processes? If you think about it, you know, one second is very fast, right? And actually ribosome can polymerize 10, on average, 10 amino acids in a second, which is really very rapid, right?', 'tokens': [50363, 1406, 345, 743, 4240, 11, 826, 11, 703, 3049, 389, 777, 7767, 30, 1002, 345, 892, 546, 340, 11, 345, 760, 11, 530, 1218, 318, 845, 3049, 11, 826, 30, 843, 1682, 12183, 418, 462, 460, 30528, 1096, 838, 11, 319, 2811, 11, 838, 23206, 17045, 287, 257, 1218, 11, 543, 318, 1107, 845, 5801, 11, 826, 30, 51113], 'temperature': 0.0, 'avg_logprob': -0.20049346923828126, 'compression_ratio': 1.6352459016393444, 'no_speech_prob': 0.20434410870075226}, {'id': 64, 'seek': 38500, 'start': 400.0, 'end': 409.0, 'text': \" As I'm talking about, even before I finish the sentence, it will be several seconds, right, perhaps. And ribosome has already made 10 amino acids linked together.\", 'tokens': [51113, 1081, 314, 1101, 3375, 546, 11, 772, 878, 314, 5461, 262, 6827, 11, 340, 481, 307, 1811, 4201, 11, 826, 11, 3737, 13, 843, 12183, 418, 462, 468, 1541, 925, 838, 23206, 17045, 6692, 1978, 13, 51563], 'temperature': 0.0, 'avg_logprob': -0.20049346923828126, 'compression_ratio': 1.6352459016393444, 'no_speech_prob': 0.20434410870075226}, {'id': 65, 'seek': 40900, 'start': 409.0, 'end': 418.0, 'text': ' And average protein is about 300 amino acids linked. So, if you compare ribosome rate that I mentioned, protein translation rate.', 'tokens': [50363, 843, 2811, 7532, 318, 546, 5867, 23206, 17045, 6692, 13, 1406, 11, 611, 345, 8996, 12183, 418, 462, 2494, 326, 314, 4750, 11, 7532, 11059, 2494, 13, 50813], 'temperature': 0.0, 'avg_logprob': -0.11784065530655231, 'compression_ratio': 1.675, 'no_speech_prob': 0.5664843320846558}, {'id': 66, 'seek': 40900, 'start': 418.0, 'end': 434.0, 'text': \" So what I'm showing you here, you don't need to memorize these numbers, but at least good to know at the level of, you know, back of the envelope number, what kind of rate that are we talking about, when you think about RNA polymerization versus amino acid polymerization.\", 'tokens': [50813, 1406, 644, 314, 1101, 4478, 345, 994, 11, 345, 836, 470, 761, 284, 16181, 1096, 777, 3146, 11, 475, 379, 1551, 922, 284, 760, 379, 262, 1241, 286, 11, 345, 760, 11, 736, 286, 262, 22878, 1271, 11, 644, 1611, 286, 2494, 326, 389, 356, 3375, 546, 11, 618, 345, 892, 546, 25897, 30528, 1634, 9051, 23206, 7408, 30528, 1634, 13, 51613], 'temperature': 0.0, 'avg_logprob': -0.11784065530655231, 'compression_ratio': 1.675, 'no_speech_prob': 0.5664843320846558}, {'id': 67, 'seek': 43400, 'start': 434.0, 'end': 450.0, 'text': ' RNA polymerization gives you mRNA from DNA, amino acid polymerization gives you protein from mRNA. So what types of enzyme that we saw last week, what does the mRNA synthesis?', 'tokens': [50363, 25897, 30528, 1634, 3607, 345, 47227, 422, 7446, 11, 23206, 7408, 30528, 1634, 3607, 345, 7532, 422, 47227, 13, 1406, 644, 3858, 286, 27679, 326, 356, 2497, 938, 1285, 11, 644, 857, 262, 47227, 21263, 30, 51163], 'temperature': 0.0, 'avg_logprob': -0.1995506167411804, 'compression_ratio': 1.4112903225806452, 'no_speech_prob': 0.42428576946258545}, {'id': 68, 'seek': 45000, 'start': 450.0, 'end': 469.0, 'text': \" So a machinery that produces mRNA, what we call these types of enzyme as, are polymerases and the RNA polymerases, right, machinery that makes DNA will be DNA polymerases, right, because it's all polymerizing the biomolecular polymers.\", 'tokens': [50363, 1406, 257, 20230, 326, 11073, 47227, 11, 644, 356, 869, 777, 3858, 286, 27679, 355, 11, 389, 30528, 1386, 290, 262, 25897, 30528, 1386, 11, 826, 11, 20230, 326, 1838, 7446, 481, 307, 7446, 30528, 1386, 11, 826, 11, 780, 340, 338, 477, 30528, 2890, 262, 27488, 2305, 10440, 7514, 11056, 13, 51313], 'temperature': 0.0, 'avg_logprob': -0.1921318428856986, 'compression_ratio': 1.5666666666666667, 'no_speech_prob': 0.4728154242038727}, {'id': 69, 'seek': 46900, 'start': 469.0, 'end': 489.0, 'text': ' If you think about the rates in terms of actual DNA replication rate is in the bacteria world on the E. coli, for instance, versus mammalian cells such as, you know, cancer cell type of cancer cell line, HeLa cells, the replication timescale are quite different, right.', 'tokens': [50363, 1002, 345, 892, 546, 262, 3965, 287, 2846, 286, 4036, 7446, 30330, 2494, 318, 287, 262, 11492, 995, 319, 262, 412, 13, 40395, 11, 329, 4554, 11, 9051, 46177, 4778, 884, 355, 11, 345, 760, 11, 4890, 2685, 2099, 286, 4890, 2685, 1627, 11, 679, 14772, 4778, 11, 262, 30330, 1661, 38765, 389, 2407, 1180, 11, 826, 13, 51363], 'temperature': 0.0, 'avg_logprob': -0.23467828381446101, 'compression_ratio': 1.5197740112994351, 'no_speech_prob': 0.4383728504180908}, {'id': 70, 'seek': 48900, 'start': 489.0, 'end': 501.0, 'text': \" But at the fundamental level, at the nucleotide level, the actual rate of transcription is, largely speaking, identical. You can make per second, if I'm taking 10 to 100 nucleotides.\", 'tokens': [50363, 887, 379, 262, 7531, 1241, 11, 379, 262, 17751, 45608, 1241, 11, 262, 4036, 2494, 286, 26955, 318, 11, 5688, 5486, 11, 10411, 13, 921, 460, 787, 583, 1218, 11, 611, 314, 1101, 2263, 838, 284, 1802, 17751, 313, 1460, 13, 50963], 'temperature': 0.0, 'avg_logprob': -0.12031697063911252, 'compression_ratio': 1.543778801843318, 'no_speech_prob': 0.4211328625679016}, {'id': 71, 'seek': 48900, 'start': 501.0, 'end': 512.0, 'text': ' But why then? Why is it that our cell division in human cells is almost a day, whereas cell division in E. coli and many bacteria are less than an hour?', 'tokens': [50963, 887, 1521, 788, 30, 4162, 318, 340, 326, 674, 2685, 7297, 287, 1692, 4778, 318, 2048, 257, 1110, 11, 9472, 2685, 7297, 287, 412, 13, 40395, 290, 867, 11492, 389, 1342, 621, 281, 1711, 30, 51513], 'temperature': 0.0, 'avg_logprob': -0.12031697063911252, 'compression_ratio': 1.543778801843318, 'no_speech_prob': 0.4211328625679016}, {'id': 72, 'seek': 51200, 'start': 512.0, 'end': 522.0, 'text': ' Why is it that DNA replication overall process takes much longer in our cells compared to E. coli?', 'tokens': [50363, 4162, 318, 340, 326, 7446, 30330, 4045, 1429, 2753, 881, 2392, 287, 674, 4778, 3688, 284, 412, 13, 40395, 30, 50863], 'temperature': 0.0, 'avg_logprob': -0.13612428307533264, 'compression_ratio': 1.0888888888888888, 'no_speech_prob': 0.3630462884902954}, {'id': 73, 'seek': 52200, 'start': 522.0, 'end': 541.0, 'text': \" Yes, certainly it's more complicated. You have to do the group checking, editing, etc. But still, the key point also is the fact that in the gene size, on average gene size, human gene is a large, much larger than bacterial gene.\", 'tokens': [50363, 3363, 11, 3729, 340, 338, 517, 8253, 13, 921, 423, 284, 466, 262, 1448, 10627, 11, 12857, 11, 3503, 13, 887, 991, 11, 262, 1994, 966, 635, 318, 262, 1109, 326, 287, 262, 9779, 2546, 11, 319, 2811, 9779, 2546, 11, 1692, 9779, 318, 257, 1588, 11, 881, 4025, 621, 23462, 9779, 13, 51313], 'temperature': 0.0, 'avg_logprob': -0.2405266845435427, 'compression_ratio': 1.506578947368421, 'no_speech_prob': 0.6162317395210266}, {'id': 74, 'seek': 54100, 'start': 541.0, 'end': 555.0, 'text': ' Can you guys think about the fundamental reason why that is the case? We talked a bit about that last week. What is it that is intrinsically, as a stark difference in the genome level between E. coli genome and human genome?', 'tokens': [50363, 1680, 345, 3730, 892, 546, 262, 7531, 1738, 1521, 326, 318, 262, 1339, 30, 775, 6619, 257, 1643, 546, 326, 938, 1285, 13, 1867, 318, 340, 326, 318, 48676, 11, 355, 257, 19278, 3580, 287, 262, 19270, 1241, 1022, 412, 13, 40395, 19270, 290, 1692, 19270, 30, 51063], 'temperature': 0.0, 'avg_logprob': -0.15858878341375612, 'compression_ratio': 1.4736842105263157, 'no_speech_prob': 0.6303730607032776}, {'id': 75, 'seek': 55500, 'start': 555.0, 'end': 570.0, 'text': \" Yeah, presence of non-coding genetic regions, so-called introns. Intons and exons, you don't have to worry about that in prokaryotic systems like E. coli or bacteria.\", 'tokens': [50363, 9425, 11, 4931, 286, 1729, 12, 66, 7656, 8513, 7652, 11, 523, 12, 7174, 9913, 684, 13, 2558, 684, 290, 409, 684, 11, 345, 836, 470, 423, 284, 5490, 546, 326, 287, 386, 74, 560, 6210, 3341, 588, 412, 13, 40395, 393, 11492, 13, 51113], 'temperature': 0.0, 'avg_logprob': -0.2008612354596456, 'compression_ratio': 1.2671755725190839, 'no_speech_prob': 0.868340015411377}, {'id': 76, 'seek': 57000, 'start': 570.0, 'end': 585.0, 'text': \" But we don't have introns in E. coli, whereas we have introns. So that means that for a given gene, even if ultimately the protein sequence is identical between E. coli protein and human protein, the actual starting genomes will be quite different.\", 'tokens': [50363, 887, 356, 836, 470, 423, 9913, 684, 287, 412, 13, 40395, 11, 9472, 356, 423, 9913, 684, 13, 1406, 326, 1724, 326, 329, 257, 1813, 9779, 11, 772, 611, 6165, 262, 7532, 8379, 318, 10411, 1022, 412, 13, 40395, 7532, 290, 1692, 7532, 11, 262, 4036, 3599, 42136, 481, 307, 2407, 1180, 13, 51113], 'temperature': 0.0, 'avg_logprob': -0.12480886358963816, 'compression_ratio': 1.5403726708074534, 'no_speech_prob': 0.8399501442909241}, {'id': 77, 'seek': 58500, 'start': 585.0, 'end': 600.0, 'text': ' Because in origin of the DNA in the human cell, we have introns and exons. Ultimately, introns are spliced out inside the nucleus, right, we talked about that last week, before you ultimately get to translated protein.', 'tokens': [50363, 4362, 287, 8159, 286, 262, 7446, 287, 262, 1692, 2685, 11, 356, 423, 9913, 684, 290, 409, 684, 13, 24199, 11, 9913, 684, 389, 4328, 3711, 503, 2641, 262, 29984, 11, 826, 11, 356, 6619, 546, 326, 938, 1285, 11, 878, 345, 6165, 651, 284, 14251, 7532, 13, 51113], 'temperature': 0.0, 'avg_logprob': -0.19976058373084435, 'compression_ratio': 1.4342105263157894, 'no_speech_prob': 0.8339758515357971}, {'id': 78, 'seek': 60000, 'start': 600.0, 'end': 615.0, 'text': ' So even if the amino acid level sequence is identical, at the DNA level, there will be introns in the human cells and eukaryotic system that will make the whole trans DNA synthesis step taking much longer.', 'tokens': [50363, 1406, 772, 611, 262, 23206, 7408, 1241, 8379, 318, 10411, 11, 379, 262, 7446, 1241, 11, 612, 481, 307, 9913, 684, 287, 262, 1692, 4778, 290, 304, 2724, 560, 6210, 1080, 326, 481, 787, 262, 2187, 1007, 7446, 21263, 2239, 2263, 881, 2392, 13, 51113], 'temperature': 0.0, 'avg_logprob': -0.11145347502173447, 'compression_ratio': 1.6221198156682028, 'no_speech_prob': 0.7764801979064941}, {'id': 79, 'seek': 60000, 'start': 615.0, 'end': 626.0, 'text': ' So replication process will be a lot longer. So that is certainly one key fundamental aspect of the eukaryotic cell, having a presence of introns.', 'tokens': [51113, 1406, 30330, 1429, 481, 307, 257, 1256, 2392, 13, 1406, 326, 318, 3729, 530, 1994, 7531, 4843, 286, 262, 304, 2724, 560, 6210, 2685, 11, 1719, 257, 4931, 286, 9913, 684, 13, 51663], 'temperature': 0.0, 'avg_logprob': -0.11145347502173447, 'compression_ratio': 1.6221198156682028, 'no_speech_prob': 0.7764801979064941}, {'id': 80, 'seek': 62600, 'start': 626.0, 'end': 637.0, 'text': \" Now, protein synthesis, which is the focus of today's discussion, as I said, the average translation rate is 10 amino acids given take per second.\", 'tokens': [50363, 2735, 11, 7532, 21263, 11, 543, 318, 262, 2962, 286, 1909, 338, 5114, 11, 355, 314, 531, 11, 262, 2811, 11059, 2494, 318, 838, 23206, 17045, 1813, 1011, 583, 1218, 13, 50913], 'temperature': 0.0, 'avg_logprob': -0.10098431327126244, 'compression_ratio': 1.6496350364963503, 'no_speech_prob': 0.24965786933898926}, {'id': 81, 'seek': 62600, 'start': 637.0, 'end': 646.0, 'text': \" So by that time, why is it that we don't have to care about these introns and exons anymore by the time we're talking about ribosome translation?\", 'tokens': [50913, 1406, 416, 326, 640, 11, 1521, 318, 340, 326, 356, 836, 470, 423, 284, 1337, 546, 777, 9913, 684, 290, 409, 684, 7471, 416, 262, 640, 356, 821, 3375, 546, 12183, 418, 462, 11059, 30, 51363], 'temperature': 0.0, 'avg_logprob': -0.10098431327126244, 'compression_ratio': 1.6496350364963503, 'no_speech_prob': 0.24965786933898926}, {'id': 82, 'seek': 62600, 'start': 646.0, 'end': 654.0, 'text': \" So the blue unit that I'm pointing out here is essentially a classical way to illustrate the ribosome because it has two subunits, a large one and a small one.\", 'tokens': [51363, 1406, 262, 4171, 4326, 326, 314, 1101, 10609, 503, 994, 318, 6986, 257, 15993, 835, 284, 19418, 262, 12183, 418, 462, 780, 340, 468, 734, 850, 41667, 11, 257, 1588, 530, 290, 257, 1402, 530, 13, 51763], 'temperature': 0.0, 'avg_logprob': -0.10098431327126244, 'compression_ratio': 1.6496350364963503, 'no_speech_prob': 0.24965786933898926}, {'id': 83, 'seek': 65400, 'start': 654.0, 'end': 661.0, 'text': ' So the green piece is the polypeptides of proteins that are being translated, are being synthesized.', 'tokens': [50363, 1406, 262, 4077, 3704, 318, 262, 7514, 431, 457, 1460, 286, 15568, 326, 389, 852, 14251, 11, 389, 852, 24983, 1143, 13, 50713], 'temperature': 0.0, 'avg_logprob': -0.08457424670835084, 'compression_ratio': 1.6310679611650485, 'no_speech_prob': 0.06005080044269562}, {'id': 84, 'seek': 65400, 'start': 661.0, 'end': 669.0, 'text': ' And then the red piece is the incoming mRNA that is coding for what amino acids should be in the protein sequence.', 'tokens': [50713, 843, 788, 262, 2266, 3704, 318, 262, 15619, 47227, 326, 318, 19617, 329, 644, 23206, 17045, 815, 307, 287, 262, 7532, 8379, 13, 51113], 'temperature': 0.0, 'avg_logprob': -0.08457424670835084, 'compression_ratio': 1.6310679611650485, 'no_speech_prob': 0.06005080044269562}, {'id': 85, 'seek': 65400, 'start': 669.0, 'end': 676.0, 'text': \" So given this, why do you think that at the point that we talk about translation, we don't have this difference anymore?\", 'tokens': [51113, 1406, 1813, 428, 11, 1521, 466, 345, 892, 326, 379, 262, 966, 326, 356, 1561, 546, 11059, 11, 356, 836, 470, 423, 428, 3580, 7471, 30, 51463], 'temperature': 0.0, 'avg_logprob': -0.08457424670835084, 'compression_ratio': 1.6310679611650485, 'no_speech_prob': 0.06005080044269562}, {'id': 86, 'seek': 67600, 'start': 676.0, 'end': 679.0, 'text': ' Yes, it is splicing. Absolutely.', 'tokens': [50363, 3363, 11, 340, 318, 4328, 6345, 13, 28416, 13, 50513], 'temperature': 0.0, 'avg_logprob': -0.18198345689212575, 'compression_ratio': 1.5449438202247192, 'no_speech_prob': 0.16806462407112122}, {'id': 87, 'seek': 67600, 'start': 679.0, 'end': 684.0, 'text': ' So mRNA that we talk about here is already a mature messenger RNA.', 'tokens': [50513, 1406, 47227, 326, 356, 1561, 546, 994, 318, 1541, 257, 15345, 31228, 25897, 13, 50763], 'temperature': 0.0, 'avg_logprob': -0.18198345689212575, 'compression_ratio': 1.5449438202247192, 'no_speech_prob': 0.16806462407112122}, {'id': 88, 'seek': 67600, 'start': 684.0, 'end': 699.0, 'text': ' Whereas the newly synthesized, the nascently synthesized, if you like, the mRNA up there at the point of the RNA polymerase doing the transcription will be the premature mRNA.', 'tokens': [50763, 28016, 262, 8308, 24983, 1143, 11, 262, 45419, 306, 24983, 1143, 11, 611, 345, 588, 11, 262, 47227, 510, 612, 379, 262, 966, 286, 262, 25897, 30528, 589, 1804, 262, 26955, 481, 307, 262, 19905, 47227, 13, 51513], 'temperature': 0.0, 'avg_logprob': -0.18198345689212575, 'compression_ratio': 1.5449438202247192, 'no_speech_prob': 0.16806462407112122}, {'id': 89, 'seek': 69900, 'start': 699.0, 'end': 709.0, 'text': ' You also tackled that question in the problem set last week, where you need to go through splicing and export, you know, decoration with poly-A tail, etc.', 'tokens': [50363, 921, 635, 35457, 326, 1808, 287, 262, 1917, 900, 938, 1285, 11, 810, 345, 761, 284, 467, 832, 4328, 6345, 290, 10784, 11, 345, 760, 11, 42050, 351, 7514, 12, 32, 7894, 11, 3503, 13, 50863], 'temperature': 0.0, 'avg_logprob': -0.15592191574421335, 'compression_ratio': 1.5454545454545454, 'no_speech_prob': 0.5966856479644775}, {'id': 90, 'seek': 69900, 'start': 709.0, 'end': 714.0, 'text': ' before you can actually get to the mature mRNA that is ready for ribosome translation.', 'tokens': [50863, 878, 345, 460, 1682, 651, 284, 262, 15345, 47227, 326, 318, 3492, 329, 12183, 418, 462, 11059, 13, 51113], 'temperature': 0.0, 'avg_logprob': -0.15592191574421335, 'compression_ratio': 1.5454545454545454, 'no_speech_prob': 0.5966856479644775}, {'id': 91, 'seek': 69900, 'start': 714.0, 'end': 718.0, 'text': \" So remember, this has to be mature mRNA. At that point, it's identical.\", 'tokens': [51113, 1406, 3505, 11, 428, 468, 284, 307, 15345, 47227, 13, 1629, 326, 966, 11, 340, 338, 10411, 13, 51313], 'temperature': 0.0, 'avg_logprob': -0.15592191574421335, 'compression_ratio': 1.5454545454545454, 'no_speech_prob': 0.5966856479644775}, {'id': 92, 'seek': 69900, 'start': 718.0, 'end': 722.0, 'text': ' Introns no longer matter because they have been spliced out.', 'tokens': [51313, 2558, 12212, 645, 2392, 2300, 780, 484, 423, 587, 4328, 3711, 503, 13, 51513], 'temperature': 0.0, 'avg_logprob': -0.15592191574421335, 'compression_ratio': 1.5454545454545454, 'no_speech_prob': 0.5966856479644775}, {'id': 93, 'seek': 72200, 'start': 722.0, 'end': 733.0, 'text': \" OK. So the take home message is there's differences in time scale, broadly speaking, and also in terms of the translation, efficiency is the same between the two systems.\", 'tokens': [50363, 7477, 13, 1406, 262, 1011, 1363, 3275, 318, 612, 338, 5400, 287, 640, 5046, 11, 18633, 5486, 11, 290, 635, 287, 2846, 286, 262, 11059, 11, 9332, 318, 262, 976, 1022, 262, 734, 3341, 13, 50913], 'temperature': 0.0, 'avg_logprob': -0.12045338227576816, 'compression_ratio': 1.5546875, 'no_speech_prob': 0.11256681382656097}, {'id': 94, 'seek': 72200, 'start': 733.0, 'end': 738.0, 'text': \" So let's think about bacterial mRNA and eukaryotic mRNA.\", 'tokens': [50913, 1406, 1309, 338, 892, 546, 23462, 47227, 290, 304, 2724, 560, 6210, 47227, 13, 51163], 'temperature': 0.0, 'avg_logprob': -0.12045338227576816, 'compression_ratio': 1.5546875, 'no_speech_prob': 0.11256681382656097}, {'id': 95, 'seek': 72200, 'start': 738.0, 'end': 744.0, 'text': \" What I'm showing you here is a domain structure, very simplified, but it gives you a generic feature.\", 'tokens': [51163, 1867, 314, 1101, 4478, 345, 994, 318, 257, 7386, 4645, 11, 845, 27009, 11, 475, 340, 3607, 345, 257, 14276, 3895, 13, 51463], 'temperature': 0.0, 'avg_logprob': -0.12045338227576816, 'compression_ratio': 1.5546875, 'no_speech_prob': 0.11256681382656097}, {'id': 96, 'seek': 72200, 'start': 744.0, 'end': 748.0, 'text': \" In terms of coding sequences, don't do the same for a given protein.\", 'tokens': [51463, 554, 2846, 286, 19617, 16311, 11, 836, 470, 466, 262, 976, 329, 257, 1813, 7532, 13, 51663], 'temperature': 0.0, 'avg_logprob': -0.12045338227576816, 'compression_ratio': 1.5546875, 'no_speech_prob': 0.11256681382656097}, {'id': 97, 'seek': 74800, 'start': 748.0, 'end': 754.0, 'text': \" Because we don't need to worry about introns anymore by the time you get to the mature mRNA.\", 'tokens': [50363, 4362, 356, 836, 470, 761, 284, 5490, 546, 9913, 684, 7471, 416, 262, 640, 345, 651, 284, 262, 15345, 47227, 13, 50663], 'temperature': 0.0, 'avg_logprob': -0.16942123517598193, 'compression_ratio': 1.5069124423963134, 'no_speech_prob': 0.26380160450935364}, {'id': 98, 'seek': 74800, 'start': 754.0, 'end': 761.0, 'text': ' But what other striking differences do you see between the two systems?', 'tokens': [50663, 887, 644, 584, 8871, 5400, 466, 345, 766, 1022, 262, 734, 3341, 30, 51013], 'temperature': 0.0, 'avg_logprob': -0.16942123517598193, 'compression_ratio': 1.5069124423963134, 'no_speech_prob': 0.26380160450935364}, {'id': 99, 'seek': 74800, 'start': 761.0, 'end': 770.0, 'text': \" Because we'll hijack quite a bit of these mRNA regulation and their individual domains and complements when we think about ribosome switch design and application,\", 'tokens': [51013, 4362, 356, 1183, 16836, 441, 2407, 257, 1643, 286, 777, 47227, 9001, 290, 511, 1981, 18209, 290, 1224, 902, 618, 356, 892, 546, 12183, 418, 462, 5078, 1486, 290, 3586, 11, 51463], 'temperature': 0.0, 'avg_logprob': -0.16942123517598193, 'compression_ratio': 1.5069124423963134, 'no_speech_prob': 0.26380160450935364}, {'id': 100, 'seek': 77000, 'start': 770.0, 'end': 776.0, 'text': ' the later part of the topic. So what do you see beyond the coding sequence?', 'tokens': [50363, 262, 1568, 636, 286, 262, 7243, 13, 1406, 644, 466, 345, 766, 3675, 262, 19617, 8379, 30, 50663], 'temperature': 0.0, 'avg_logprob': -0.18111299857115135, 'compression_ratio': 1.6647727272727273, 'no_speech_prob': 0.5201756358146667}, {'id': 101, 'seek': 77000, 'start': 776.0, 'end': 784.0, 'text': ' Coding sequence is the mRNA that will code for your amino acid sequence on the protein level.', 'tokens': [50663, 327, 7656, 8379, 318, 262, 47227, 326, 481, 2438, 329, 534, 23206, 7408, 8379, 319, 262, 7532, 1241, 13, 51063], 'temperature': 0.0, 'avg_logprob': -0.18111299857115135, 'compression_ratio': 1.6647727272727273, 'no_speech_prob': 0.5201756358146667}, {'id': 102, 'seek': 77000, 'start': 784.0, 'end': 789.0, 'text': ' You see untranslated region. So this is upstream untranslated region.', 'tokens': [51063, 921, 766, 1418, 26084, 17249, 3814, 13, 1406, 428, 318, 28717, 1418, 26084, 17249, 3814, 13, 51313], 'temperature': 0.0, 'avg_logprob': -0.18111299857115135, 'compression_ratio': 1.6647727272727273, 'no_speech_prob': 0.5201756358146667}, {'id': 103, 'seek': 77000, 'start': 789.0, 'end': 795.0, 'text': \" So 5' UTR, so untranslated region and downstream UTR.\", 'tokens': [51313, 1406, 642, 6, 471, 5446, 11, 523, 1418, 26084, 17249, 3814, 290, 33218, 471, 5446, 13, 51613], 'temperature': 0.0, 'avg_logprob': -0.18111299857115135, 'compression_ratio': 1.6647727272727273, 'no_speech_prob': 0.5201756358146667}, {'id': 104, 'seek': 79500, 'start': 795.0, 'end': 799.0, 'text': ' And those are present in both cases.', 'tokens': [50363, 843, 883, 389, 1944, 287, 1111, 2663, 13, 50563], 'temperature': 0.0, 'avg_logprob': -0.18963532919412132, 'compression_ratio': 1.6085106382978724, 'no_speech_prob': 0.08599299937486649}, {'id': 105, 'seek': 79500, 'start': 799.0, 'end': 805.0, 'text': \" Those components are essential for gene regulation and stability of mRNA. We'll come to that.\", 'tokens': [50563, 5845, 6805, 389, 6393, 329, 9779, 9001, 290, 10159, 286, 47227, 13, 775, 1183, 1282, 284, 326, 13, 50863], 'temperature': 0.0, 'avg_logprob': -0.18963532919412132, 'compression_ratio': 1.6085106382978724, 'no_speech_prob': 0.08599299937486649}, {'id': 106, 'seek': 79500, 'start': 805.0, 'end': 809.0, 'text': ' And that essentially happens in both bacteria and eukaryotic world.', 'tokens': [50863, 843, 326, 6986, 4325, 287, 1111, 11492, 290, 304, 2724, 560, 6210, 995, 13, 51063], 'temperature': 0.0, 'avg_logprob': -0.18963532919412132, 'compression_ratio': 1.6085106382978724, 'no_speech_prob': 0.08599299937486649}, {'id': 107, 'seek': 79500, 'start': 809.0, 'end': 814.0, 'text': \" But you see additional units such as poly-A tail and 5' prime capping.\", 'tokens': [51063, 887, 345, 766, 3224, 4991, 884, 355, 7514, 12, 32, 7894, 290, 642, 6, 6994, 1275, 2105, 13, 51313], 'temperature': 0.0, 'avg_logprob': -0.18963532919412132, 'compression_ratio': 1.6085106382978724, 'no_speech_prob': 0.08599299937486649}, {'id': 108, 'seek': 79500, 'start': 814.0, 'end': 823.0, 'text': ' That is absolutely essential to introduce this poly-A tail tagging and then cap modification on the 5 prime.', 'tokens': [51313, 1320, 318, 5543, 6393, 284, 10400, 428, 7514, 12, 32, 7894, 49620, 290, 788, 1451, 17613, 319, 262, 642, 6994, 13, 51763], 'temperature': 0.0, 'avg_logprob': -0.18963532919412132, 'compression_ratio': 1.6085106382978724, 'no_speech_prob': 0.08599299937486649}, {'id': 109, 'seek': 82300, 'start': 823.0, 'end': 831.0, 'text': ' Fourth, eukaryotic mRNA, because when you think about eukaryotic mRNA, they are first produced in the nucleus,', 'tokens': [50363, 15692, 11, 304, 2724, 560, 6210, 47227, 11, 780, 618, 345, 892, 546, 304, 2724, 560, 6210, 47227, 11, 484, 389, 717, 4635, 287, 262, 29984, 11, 50763], 'temperature': 0.0, 'avg_logprob': -0.13216607466987942, 'compression_ratio': 1.5086705202312138, 'no_speech_prob': 0.20860248804092407}, {'id': 110, 'seek': 82300, 'start': 831.0, 'end': 842.0, 'text': ' but they need to be chaperoned and exported in a safe manner to the cytoplasm where they will meet ribosome such that they can synthesize the protein.', 'tokens': [50763, 475, 484, 761, 284, 307, 442, 2136, 12004, 290, 29050, 287, 257, 3338, 5642, 284, 262, 3075, 4852, 75, 8597, 810, 484, 481, 1826, 12183, 418, 462, 884, 326, 484, 460, 24983, 1096, 262, 7532, 13, 51313], 'temperature': 0.0, 'avg_logprob': -0.13216607466987942, 'compression_ratio': 1.5086705202312138, 'no_speech_prob': 0.20860248804092407}, {'id': 111, 'seek': 84200, 'start': 843.0, 'end': 850.0, 'text': ' If you have mRNA produced in the nucleus that are not stable and protected by nucleases degradation,', 'tokens': [50413, 1002, 345, 423, 47227, 4635, 287, 262, 29984, 326, 389, 407, 8245, 290, 6861, 416, 17751, 1386, 27486, 11, 50763], 'temperature': 0.0, 'avg_logprob': -0.16692736377454784, 'compression_ratio': 1.5693069306930694, 'no_speech_prob': 0.7356226444244385}, {'id': 112, 'seek': 84200, 'start': 850.0, 'end': 856.0, 'text': ' we talked a lot about RNA nucleases last lecture as a means of therapeutic mechanism.', 'tokens': [50763, 356, 6619, 257, 1256, 546, 25897, 17751, 1386, 938, 19143, 355, 257, 1724, 286, 21546, 9030, 13, 51063], 'temperature': 0.0, 'avg_logprob': -0.16692736377454784, 'compression_ratio': 1.5693069306930694, 'no_speech_prob': 0.7356226444244385}, {'id': 113, 'seek': 84200, 'start': 856.0, 'end': 865.0, 'text': ' Both nucleases could essentially chop up the RNA that is newly synthesized in the nucleus before it can even get to the cytoplasm.', 'tokens': [51063, 5747, 17751, 1386, 714, 6986, 30506, 510, 262, 25897, 326, 318, 8308, 24983, 1143, 287, 262, 29984, 878, 340, 460, 772, 651, 284, 262, 3075, 4852, 75, 8597, 13, 51513], 'temperature': 0.0, 'avg_logprob': -0.16692736377454784, 'compression_ratio': 1.5693069306930694, 'no_speech_prob': 0.7356226444244385}, {'id': 114, 'seek': 86500, 'start': 865.0, 'end': 891.0, 'text': ' So these accessories, 5 prime capping and poly-A tail tagging at the mRNA level is critical to stabilize and protect the eukaryotic mRNA such that they can be recognized by proteins that will bind to these units and actually chaperone them, escort them if you like, for mRNA export from the nucleus to the cytoplasm for protein translation.', 'tokens': [50363, 1406, 777, 18199, 11, 642, 6994, 1275, 2105, 290, 7514, 12, 32, 7894, 49620, 379, 262, 47227, 1241, 318, 4688, 284, 32411, 290, 1805, 262, 304, 2724, 560, 6210, 47227, 884, 326, 484, 460, 307, 8018, 416, 15568, 326, 481, 11007, 284, 777, 4991, 290, 1682, 442, 2136, 505, 606, 11, 27675, 606, 611, 345, 588, 11, 329, 47227, 10784, 422, 262, 29984, 284, 262, 3075, 4852, 75, 8597, 329, 7532, 11059, 13, 51663], 'temperature': 0.0, 'avg_logprob': -0.13183381959989474, 'compression_ratio': 1.596244131455399, 'no_speech_prob': 0.26294776797294617}, {'id': 115, 'seek': 89100, 'start': 891.0, 'end': 898.0, 'text': ' Keep in mind of this kind of segregation of compartments that makes the system more complex.', 'tokens': [50363, 9175, 287, 2000, 286, 428, 1611, 286, 26718, 286, 552, 32514, 326, 1838, 262, 1080, 517, 3716, 13, 50713], 'temperature': 0.0, 'avg_logprob': -0.20717738373110992, 'compression_ratio': 1.5393700787401574, 'no_speech_prob': 0.40725141763687134}, {'id': 116, 'seek': 89100, 'start': 898.0, 'end': 909.0, 'text': ' That means you require more factors to stabilize these mRNAs, protect these mRNA, chaperone and export them safely to the cytoplasm.', 'tokens': [50713, 1320, 1724, 345, 2421, 517, 5087, 284, 32411, 777, 285, 42336, 1722, 11, 1805, 777, 47227, 11, 442, 2136, 505, 290, 10784, 606, 11512, 284, 262, 3075, 4852, 75, 8597, 13, 51263], 'temperature': 0.0, 'avg_logprob': -0.20717738373110992, 'compression_ratio': 1.5393700787401574, 'no_speech_prob': 0.40725141763687134}, {'id': 117, 'seek': 89100, 'start': 909.0, 'end': 912.0, 'text': \" So that's what you're seeing here.\", 'tokens': [51263, 1406, 326, 338, 644, 345, 821, 4379, 994, 13, 51413], 'temperature': 0.0, 'avg_logprob': -0.20717738373110992, 'compression_ratio': 1.5393700787401574, 'no_speech_prob': 0.40725141763687134}, {'id': 118, 'seek': 89100, 'start': 912.0, 'end': 920.0, 'text': \" Sub-cellular loca will already talk about it. What I've just discussed, you can also read further in your own time in these texts.\", 'tokens': [51413, 3834, 12, 3846, 934, 1179, 64, 481, 1541, 1561, 546, 340, 13, 1867, 314, 1053, 655, 6693, 11, 345, 460, 635, 1100, 2252, 287, 534, 898, 640, 287, 777, 13399, 13, 51813], 'temperature': 0.0, 'avg_logprob': -0.20717738373110992, 'compression_ratio': 1.5393700787401574, 'no_speech_prob': 0.40725141763687134}, {'id': 119, 'seek': 92000, 'start': 920.0, 'end': 932.0, 'text': \" But let's go through a little bit more of what allows a ribosome to know you have a long piece of mRNA starting from the genome that's coming along.\", 'tokens': [50363, 887, 1309, 338, 467, 832, 257, 1310, 1643, 517, 286, 644, 3578, 257, 12183, 418, 462, 284, 760, 345, 423, 257, 890, 3704, 286, 47227, 3599, 422, 262, 19270, 326, 338, 2406, 1863, 13, 50963], 'temperature': 0.0, 'avg_logprob': -0.16349092343958413, 'compression_ratio': 1.6650717703349283, 'no_speech_prob': 0.32910171151161194}, {'id': 120, 'seek': 92000, 'start': 932.0, 'end': 944.0, 'text': ' How does ribosome know when to start translating, translating and synthesizing the amino acid, if you make this polymeric chain for a protein, when to start this process and when to end this process?', 'tokens': [50963, 1374, 857, 12183, 418, 462, 760, 618, 284, 923, 34665, 11, 34665, 290, 24983, 2890, 262, 23206, 7408, 11, 611, 345, 787, 428, 7514, 946, 6333, 329, 257, 7532, 11, 618, 284, 923, 428, 1429, 290, 618, 284, 886, 428, 1429, 30, 51563], 'temperature': 0.0, 'avg_logprob': -0.16349092343958413, 'compression_ratio': 1.6650717703349283, 'no_speech_prob': 0.32910171151161194}, {'id': 121, 'seek': 94400, 'start': 944.0, 'end': 955.0, 'text': ' So there are specific codes that you will be implemented in your mRNA in nature that allows a ribosome to signal that this must be a star site. Star codon is always AUG.', 'tokens': [50363, 1406, 612, 389, 2176, 12416, 326, 345, 481, 307, 9177, 287, 534, 47227, 287, 3450, 326, 3578, 257, 12183, 418, 462, 284, 6737, 326, 428, 1276, 307, 257, 3491, 2524, 13, 2907, 14873, 261, 318, 1464, 317, 7340, 13, 50913], 'temperature': 0.0, 'avg_logprob': -0.1386324988471137, 'compression_ratio': 1.7718446601941749, 'no_speech_prob': 0.7562346458435059}, {'id': 122, 'seek': 94400, 'start': 955.0, 'end': 967.0, 'text': ' And then there will be the end site, there will be a star codon in nature, there are three star codons and these three nucleotides will signal for where the star site and the end point should be.', 'tokens': [50913, 843, 788, 612, 481, 307, 262, 886, 2524, 11, 612, 481, 307, 257, 3491, 14873, 261, 287, 3450, 11, 612, 389, 1115, 3491, 14873, 684, 290, 777, 1115, 17751, 313, 1460, 481, 6737, 329, 810, 262, 3491, 2524, 290, 262, 886, 966, 815, 307, 13, 51513], 'temperature': 0.0, 'avg_logprob': -0.1386324988471137, 'compression_ratio': 1.7718446601941749, 'no_speech_prob': 0.7562346458435059}, {'id': 123, 'seek': 96700, 'start': 967.0, 'end': 981.0, 'text': \" Because if you think about it, these are, you know, if they are altered in any way, you'll have a protein that are either truncated, i.e. too short, shorter than you desire, or maybe start in the long side.\", 'tokens': [50363, 4362, 611, 345, 892, 546, 340, 11, 777, 389, 11, 345, 760, 11, 611, 484, 389, 14294, 287, 597, 835, 11, 345, 1183, 423, 257, 7532, 326, 389, 2035, 40122, 515, 11, 1312, 13, 68, 13, 1165, 1790, 11, 12238, 621, 345, 6227, 11, 393, 3863, 923, 287, 262, 890, 1735, 13, 51063], 'temperature': 0.0, 'avg_logprob': -0.11341676046681959, 'compression_ratio': 1.647887323943662, 'no_speech_prob': 0.42656153440475464}, {'id': 124, 'seek': 96700, 'start': 981.0, 'end': 990.0, 'text': ' So these stop and start codons and signals are critical for ribosomes to understand where the translation should be started and should be ended.', 'tokens': [51063, 1406, 777, 2245, 290, 923, 14873, 684, 290, 10425, 389, 4688, 329, 12183, 418, 2586, 284, 1833, 810, 262, 11059, 815, 307, 2067, 290, 815, 307, 4444, 13, 51513], 'temperature': 0.0, 'avg_logprob': -0.11341676046681959, 'compression_ratio': 1.647887323943662, 'no_speech_prob': 0.42656153440475464}, {'id': 125, 'seek': 99000, 'start': 990.0, 'end': 998.0, 'text': \" So let's take a look at this in a sort of an overview level from the whole process with this illustration.\", 'tokens': [50363, 1406, 1309, 338, 1011, 257, 804, 379, 428, 287, 257, 3297, 286, 281, 16700, 1241, 422, 262, 2187, 1429, 351, 428, 20936, 13, 50763], 'temperature': 0.0, 'avg_logprob': -0.12582741086445157, 'compression_ratio': 1.6145251396648044, 'no_speech_prob': 0.3834340274333954}, {'id': 126, 'seek': 99000, 'start': 998.0, 'end': 1009.0, 'text': ' Many of you probably have come across chemical polymerization, perhaps in high school, and you know, a lot of polymerization processes in chemistry and industry, chemical processing.', 'tokens': [50763, 4650, 286, 345, 2192, 423, 1282, 1973, 5931, 30528, 1634, 11, 3737, 287, 1029, 1524, 11, 290, 345, 760, 11, 257, 1256, 286, 30528, 1634, 7767, 287, 16585, 290, 2831, 11, 5931, 7587, 13, 51313], 'temperature': 0.0, 'avg_logprob': -0.12582741086445157, 'compression_ratio': 1.6145251396648044, 'no_speech_prob': 0.3834340274333954}, {'id': 127, 'seek': 100900, 'start': 1009.0, 'end': 1019.0, 'text': ' Biology in many cases work the same way. In any polymerization process, what do you have? There are typically three steps in any kind of polymerization.', 'tokens': [50363, 24698, 287, 867, 2663, 670, 262, 976, 835, 13, 554, 597, 30528, 1634, 1429, 11, 644, 466, 345, 423, 30, 1318, 389, 6032, 1115, 4831, 287, 597, 1611, 286, 30528, 1634, 13, 50863], 'temperature': 0.0, 'avg_logprob': -0.2726984715116197, 'compression_ratio': 1.6051282051282052, 'no_speech_prob': 0.8261743187904358}, {'id': 128, 'seek': 100900, 'start': 1019.0, 'end': 1030.0, 'text': \" Sometimes they change the activities, that's how we would encounter in USA is we encounter chemical polymerization. There are always three steps. What are they?\", 'tokens': [50863, 8975, 484, 1487, 262, 4568, 11, 326, 338, 703, 356, 561, 8791, 287, 4916, 318, 356, 8791, 5931, 30528, 1634, 13, 1318, 389, 1464, 1115, 4831, 13, 1867, 389, 484, 30, 51413], 'temperature': 0.0, 'avg_logprob': -0.2726984715116197, 'compression_ratio': 1.6051282051282052, 'no_speech_prob': 0.8261743187904358}, {'id': 129, 'seek': 103000, 'start': 1030.0, 'end': 1042.0, 'text': \" You have to initiate, you have to elongate, process, and then you have to terminate. Biology does the same thing, nature does the same thing. There's initiation step, elongation step, and termination step.\", 'tokens': [50363, 921, 423, 284, 22118, 11, 345, 423, 284, 39696, 378, 11, 1429, 11, 290, 788, 345, 423, 284, 23654, 13, 24698, 857, 262, 976, 1517, 11, 3450, 857, 262, 976, 1517, 13, 1318, 338, 30122, 2239, 11, 39696, 341, 2239, 11, 290, 19883, 2239, 13, 50963], 'temperature': 0.0, 'avg_logprob': -0.15458863258361816, 'compression_ratio': 1.7962962962962963, 'no_speech_prob': 0.2026246190071106}, {'id': 130, 'seek': 103000, 'start': 1042.0, 'end': 1057.0, 'text': \" So let's take a walkthrough of these individual steps. So what I'm showing you here is mRNA, 5' and 3', and you can see the stop codon, and stop codon is one of the stop codons here.\", 'tokens': [50963, 1406, 1309, 338, 1011, 257, 2513, 9579, 286, 777, 1981, 4831, 13, 1406, 644, 314, 1101, 4478, 345, 994, 318, 47227, 11, 642, 6, 290, 513, 3256, 290, 345, 460, 766, 262, 2245, 14873, 261, 11, 290, 2245, 14873, 261, 318, 530, 286, 262, 2245, 14873, 684, 994, 13, 51713], 'temperature': 0.0, 'avg_logprob': -0.15458863258361816, 'compression_ratio': 1.7962962962962963, 'no_speech_prob': 0.2026246190071106}, {'id': 131, 'seek': 105700, 'start': 1058.0, 'end': 1072.0, 'text': ' And ribosome essentially is made up of sort of two sub-major subunits, one large and one small. So not just having an mRNA and this ribosome machinery, obviously something has to bring in your individual amino acid.', 'tokens': [50413, 843, 12183, 418, 462, 6986, 318, 925, 510, 286, 3297, 286, 734, 850, 12, 22478, 850, 41667, 11, 530, 1588, 290, 530, 1402, 13, 1406, 407, 655, 1719, 281, 47227, 290, 428, 12183, 418, 462, 20230, 11, 6189, 1223, 468, 284, 2222, 287, 534, 1981, 23206, 7408, 13, 51113], 'temperature': 0.0, 'avg_logprob': -0.18110293608445388, 'compression_ratio': 1.396103896103896, 'no_speech_prob': 0.18400055170059204}, {'id': 132, 'seek': 107200, 'start': 1073.0, 'end': 1096.0, 'text': ' So how does nature do that in all living systems? And it turns out that there is this additional type of RNA, still nucleic acid, called transfer RNA, tRNA, which serves as an adapter to bring in the precise amino acid shown here in the purple color, to actually allow this polymerization process to happen in a coded manner.', 'tokens': [50413, 1406, 703, 857, 3450, 466, 326, 287, 477, 2877, 3341, 30, 843, 340, 4962, 503, 326, 612, 318, 428, 3224, 2099, 286, 25897, 11, 991, 17751, 291, 7408, 11, 1444, 4351, 25897, 11, 256, 27204, 11, 543, 9179, 355, 281, 21302, 284, 2222, 287, 262, 7141, 23206, 7408, 3402, 994, 287, 262, 14032, 3124, 11, 284, 1682, 1249, 428, 30528, 1634, 1429, 284, 1645, 287, 257, 30817, 5642, 13, 51563], 'temperature': 0.0, 'avg_logprob': -0.0989885395520354, 'compression_ratio': 1.5330188679245282, 'no_speech_prob': 0.4177020192146301}, {'id': 133, 'seek': 109600, 'start': 1097.0, 'end': 1106.0, 'text': ' I said coded manner because which amino acid to be brought in will be defined by the actual nucleotide sequence that is present along your mRNA.', 'tokens': [50413, 314, 531, 30817, 5642, 780, 543, 23206, 7408, 284, 307, 3181, 287, 481, 307, 5447, 416, 262, 4036, 17751, 45608, 8379, 326, 318, 1944, 1863, 534, 47227, 13, 50863], 'temperature': 0.0, 'avg_logprob': -0.15413781368371213, 'compression_ratio': 1.6602316602316602, 'no_speech_prob': 0.23855355381965637}, {'id': 134, 'seek': 109600, 'start': 1108.0, 'end': 1125.0, 'text': ' How do you think this position happens? So mRNA will have different nucleosides here, and I could have tRNA bringing in one amino acid at a time. How do I know how this system, how can you encode for the fact that if I hit this particular codon, I will bring in the correct amino acid?', 'tokens': [50963, 1374, 466, 345, 892, 428, 2292, 4325, 30, 1406, 47227, 481, 423, 1180, 17751, 418, 1460, 994, 11, 290, 314, 714, 423, 256, 27204, 6079, 287, 530, 23206, 7408, 379, 257, 640, 13, 1374, 466, 314, 760, 703, 428, 1080, 11, 703, 460, 345, 37773, 329, 262, 1109, 326, 611, 314, 2277, 428, 1948, 14873, 261, 11, 314, 481, 2222, 287, 262, 3376, 23206, 7408, 30, 51813], 'temperature': 0.0, 'avg_logprob': -0.15413781368371213, 'compression_ratio': 1.6602316602316602, 'no_speech_prob': 0.23855355381965637}]\n"
     ]
    }
   ],
   "source": [
    "transcription = joblib.load('output/transcription_google_colab.jl')\n",
    "print(transcription[0].keys())\n",
    "print(transcription[0]['segments'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   id  seek  start  end                                            text   \n",
      "0   0     0    0.0  1.0                                  Thank you all.  \\\n",
      "1   1     0    1.0  2.0                                 Good afternoon.   \n",
      "2   2     0    2.0  3.0                         How are you guys doing?   \n",
      "3   3     0    3.0  4.0                                   It looks fun.   \n",
      "4   4     0    4.0  7.0   How many fans have done themselves last week?   \n",
      "\n",
      "                                              tokens  temperature   \n",
      "0                 [50363, 6952, 345, 477, 13, 50413]          0.0  \\\n",
      "1                     [50413, 4599, 6672, 13, 50463]          0.0   \n",
      "2     [50463, 1374, 389, 345, 3730, 1804, 30, 50513]          0.0   \n",
      "3                [50513, 632, 3073, 1257, 13, 50563]          0.0   \n",
      "4  [50563, 1374, 867, 3296, 423, 1760, 2405, 938,...          0.0   \n",
      "\n",
      "   avg_logprob  compression_ratio  no_speech_prob  \n",
      "0    -0.454801           1.611354        0.274894  \n",
      "1    -0.454801           1.611354        0.274894  \n",
      "2    -0.454801           1.611354        0.274894  \n",
      "3    -0.454801           1.611354        0.274894  \n",
      "4    -0.454801           1.611354        0.274894  \n",
      " Many of you probably have come across chemical polymerization, perhaps in high school, and you know, a lot of polymerization processes in chemistry and industry, chemical processing.\n",
      " Biology in many cases work the same way. In any polymerization process, what do you have? There are typically three steps in any kind of polymerization.\n",
      " Sometimes they change the activities, that's how we would encounter in USA is we encounter chemical polymerization. There are always three steps. What are they?\n",
      " You have to initiate, you have to elongate, process, and then you have to terminate. Biology does the same thing, nature does the same thing. There's initiation step, elongation step, and termination step.\n",
      " So let's take a walkthrough of these individual steps. So what I'm showing you here is mRNA, 5' and 3', and you can see the stop codon, and stop codon is one of the stop codons here.\n",
      " And ribosome essentially is made up of sort of two sub-major subunits, one large and one small. So not just having an mRNA and this ribosome machinery, obviously something has to bring in your individual amino acid.\n",
      " So how does nature do that in all living systems? And it turns out that there is this additional type of RNA, still nucleic acid, called transfer RNA, tRNA, which serves as an adapter to bring in the precise amino acid shown here in the purple color, to actually allow this polymerization process to happen in a coded manner.\n",
      " I said coded manner because which amino acid to be brought in will be defined by the actual nucleotide sequence that is present along your mRNA.\n",
      " How do you think this position happens? So mRNA will have different nucleosides here, and I could have tRNA bringing in one amino acid at a time. How do I know how this system, how can you encode for the fact that if I hit this particular codon, I will bring in the correct amino acid?\n"
     ]
    }
   ],
   "source": [
    "subclip_0_df = pd.DataFrame(transcription[0]['segments'])\n",
    "print(subclip_0_df.head())\n",
    "for i in range(-9,0):\n",
    "    print(subclip_0_df['text'].iloc[i])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " How do you think this position happens?\n",
      " So mRNA will have different nucleoside fields\n",
      " and I could have tRNA bringing in one amino acid at a time.\n",
      " How do I know how does the system,\n",
      " how can you encode for the fact that,\n",
      " if I hit this particular code off,\n",
      " I will bring in the correct amino acid.\n",
      " There's chemistry behind it.\n",
      " What is this position mechanism?\n",
      " Yeah.\n"
     ]
    }
   ],
   "source": [
    "subclip_1_df = pd.DataFrame(transcription[1]['segments'])\n",
    "for i in range(10):\n",
    "    print(subclip_1_df['text'].iloc[i])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>path</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1128</td>\n",
       "      <td>output/subclip_0_0_1128.mp3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1108</td>\n",
       "      <td>2246</td>\n",
       "      <td>output/subclip_1_1108_2246.mp3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>2226</td>\n",
       "      <td>3364</td>\n",
       "      <td>output/subclip_2_2226_3364.mp3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>3344</td>\n",
       "      <td>4482</td>\n",
       "      <td>output/subclip_3_3344_4482.mp3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>4462</td>\n",
       "      <td>5587</td>\n",
       "      <td>output/subclip_4_4462_5587.mp3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  index  start   end                            path\n",
       "0     0      0  1128     output/subclip_0_0_1128.mp3\n",
       "1     1   1108  2246  output/subclip_1_1108_2246.mp3\n",
       "2     2   2226  3364  output/subclip_2_2226_3364.mp3\n",
       "3     3   3344  4482  output/subclip_3_3344_4482.mp3\n",
       "4     4   4462  5587  output/subclip_4_4462_5587.mp3"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subclips_list = joblib.load('output/subclips_list.jl')\n",
    "subclip_info_df = pd.DataFrame(subclips_list)\n",
    "subclip_info_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "subclip_0_df['subclip_index'] = subclip_info_df['index'].iloc[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>seek</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>text</th>\n",
       "      <th>tokens</th>\n",
       "      <th>temperature</th>\n",
       "      <th>avg_logprob</th>\n",
       "      <th>compression_ratio</th>\n",
       "      <th>no_speech_prob</th>\n",
       "      <th>subclip_index</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>Thank you all.</td>\n",
       "      <td>[50363, 6952, 345, 477, 13, 50413]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.454801</td>\n",
       "      <td>1.611354</td>\n",
       "      <td>0.274894</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.00</td>\n",
       "      <td>2.00</td>\n",
       "      <td>Good afternoon.</td>\n",
       "      <td>[50413, 4599, 6672, 13, 50463]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.454801</td>\n",
       "      <td>1.611354</td>\n",
       "      <td>0.274894</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2.00</td>\n",
       "      <td>3.00</td>\n",
       "      <td>How are you guys doing?</td>\n",
       "      <td>[50463, 1374, 389, 345, 3730, 1804, 30, 50513]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.454801</td>\n",
       "      <td>1.611354</td>\n",
       "      <td>0.274894</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>3.00</td>\n",
       "      <td>4.00</td>\n",
       "      <td>It looks fun.</td>\n",
       "      <td>[50513, 632, 3073, 1257, 13, 50563]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.454801</td>\n",
       "      <td>1.611354</td>\n",
       "      <td>0.274894</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>4.00</td>\n",
       "      <td>7.00</td>\n",
       "      <td>How many fans have done themselves last week?</td>\n",
       "      <td>[50563, 1374, 867, 3296, 423, 1760, 2405, 938,...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.454801</td>\n",
       "      <td>1.611354</td>\n",
       "      <td>0.274894</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1480</th>\n",
       "      <td>162</td>\n",
       "      <td>108472</td>\n",
       "      <td>5553.12</td>\n",
       "      <td>5558.56</td>\n",
       "      <td>one more class to see you guys, but there wil...</td>\n",
       "      <td>[50683, 530, 517, 1398, 284, 766, 345, 3730, 1...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.208055</td>\n",
       "      <td>1.514894</td>\n",
       "      <td>0.001780</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1481</th>\n",
       "      <td>163</td>\n",
       "      <td>108472</td>\n",
       "      <td>5558.56</td>\n",
       "      <td>5567.44</td>\n",
       "      <td>game-changing systems that you'll learn next ...</td>\n",
       "      <td>[50955, 983, 12, 22954, 3341, 326, 345, 1183, ...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.208055</td>\n",
       "      <td>1.514894</td>\n",
       "      <td>0.001780</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1482</th>\n",
       "      <td>164</td>\n",
       "      <td>108472</td>\n",
       "      <td>5567.44</td>\n",
       "      <td>5571.68</td>\n",
       "      <td>If you have questions, bring to us. And if yo...</td>\n",
       "      <td>[51399, 1002, 345, 423, 2683, 11, 2222, 284, 5...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.208055</td>\n",
       "      <td>1.514894</td>\n",
       "      <td>0.001780</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1483</th>\n",
       "      <td>165</td>\n",
       "      <td>110968</td>\n",
       "      <td>5571.68</td>\n",
       "      <td>5576.64</td>\n",
       "      <td>you will find problems as really quite straig...</td>\n",
       "      <td>[50363, 345, 481, 1064, 2761, 355, 1107, 2407,...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.245308</td>\n",
       "      <td>1.240310</td>\n",
       "      <td>0.044770</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1484</th>\n",
       "      <td>166</td>\n",
       "      <td>110968</td>\n",
       "      <td>5576.64</td>\n",
       "      <td>5584.00</td>\n",
       "      <td>you guys again next Monday. Okay. Good. Have ...</td>\n",
       "      <td>[50611, 345, 3730, 757, 1306, 3321, 13, 16805,...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.245308</td>\n",
       "      <td>1.240310</td>\n",
       "      <td>0.044770</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1485 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       id    seek    start      end   \n",
       "0       0       0     0.00     1.00  \\\n",
       "1       1       0     1.00     2.00   \n",
       "2       2       0     2.00     3.00   \n",
       "3       3       0     3.00     4.00   \n",
       "4       4       0     4.00     7.00   \n",
       "...   ...     ...      ...      ...   \n",
       "1480  162  108472  5553.12  5558.56   \n",
       "1481  163  108472  5558.56  5567.44   \n",
       "1482  164  108472  5567.44  5571.68   \n",
       "1483  165  110968  5571.68  5576.64   \n",
       "1484  166  110968  5576.64  5584.00   \n",
       "\n",
       "                                                   text   \n",
       "0                                        Thank you all.  \\\n",
       "1                                       Good afternoon.   \n",
       "2                               How are you guys doing?   \n",
       "3                                         It looks fun.   \n",
       "4         How many fans have done themselves last week?   \n",
       "...                                                 ...   \n",
       "1480   one more class to see you guys, but there wil...   \n",
       "1481   game-changing systems that you'll learn next ...   \n",
       "1482   If you have questions, bring to us. And if yo...   \n",
       "1483   you will find problems as really quite straig...   \n",
       "1484   you guys again next Monday. Okay. Good. Have ...   \n",
       "\n",
       "                                                 tokens  temperature   \n",
       "0                    [50363, 6952, 345, 477, 13, 50413]          0.0  \\\n",
       "1                        [50413, 4599, 6672, 13, 50463]          0.0   \n",
       "2        [50463, 1374, 389, 345, 3730, 1804, 30, 50513]          0.0   \n",
       "3                   [50513, 632, 3073, 1257, 13, 50563]          0.0   \n",
       "4     [50563, 1374, 867, 3296, 423, 1760, 2405, 938,...          0.0   \n",
       "...                                                 ...          ...   \n",
       "1480  [50683, 530, 517, 1398, 284, 766, 345, 3730, 1...          0.0   \n",
       "1481  [50955, 983, 12, 22954, 3341, 326, 345, 1183, ...          0.0   \n",
       "1482  [51399, 1002, 345, 423, 2683, 11, 2222, 284, 5...          0.0   \n",
       "1483  [50363, 345, 481, 1064, 2761, 355, 1107, 2407,...          0.0   \n",
       "1484  [50611, 345, 3730, 757, 1306, 3321, 13, 16805,...          0.0   \n",
       "\n",
       "      avg_logprob  compression_ratio  no_speech_prob subclip_index  \n",
       "0       -0.454801           1.611354        0.274894             0  \n",
       "1       -0.454801           1.611354        0.274894             0  \n",
       "2       -0.454801           1.611354        0.274894             0  \n",
       "3       -0.454801           1.611354        0.274894             0  \n",
       "4       -0.454801           1.611354        0.274894             0  \n",
       "...           ...                ...             ...           ...  \n",
       "1480    -0.208055           1.514894        0.001780             4  \n",
       "1481    -0.208055           1.514894        0.001780             4  \n",
       "1482    -0.208055           1.514894        0.001780             4  \n",
       "1483    -0.245308           1.240310        0.044770             4  \n",
       "1484    -0.245308           1.240310        0.044770             4  \n",
       "\n",
       "[1485 rows x 11 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subclip_0_df['start'] += subclip_info_df['start'].iloc[0]\n",
    "subclip_0_df['end'] += subclip_info_df['start'].iloc[0]\n",
    "\n",
    "for i in range(1, len(transcription)):\n",
    "    subclip_i_df = pd.DataFrame(transcription[i]['segments'])\n",
    "    subclip_i_df['subclip_index'] = subclip_info_df['index'].iloc[i]\n",
    "    subclip_i_df['start'] += subclip_info_df['start'].iloc[i]\n",
    "    subclip_i_df['end'] += subclip_info_df['start'].iloc[i]\n",
    "    subclip_0_df = pd.concat([subclip_0_df, subclip_i_df], ignore_index=True)\n",
    "subclip_0_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>seek</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>text</th>\n",
       "      <th>tokens</th>\n",
       "      <th>temperature</th>\n",
       "      <th>avg_logprob</th>\n",
       "      <th>compression_ratio</th>\n",
       "      <th>no_speech_prob</th>\n",
       "      <th>subclip_index</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>133</th>\n",
       "      <td>133</td>\n",
       "      <td>109600</td>\n",
       "      <td>1097.00</td>\n",
       "      <td>1106.00</td>\n",
       "      <td>I said coded manner because which amino acid ...</td>\n",
       "      <td>[50413, 314, 531, 30817, 5642, 780, 543, 23206...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.154138</td>\n",
       "      <td>1.660232</td>\n",
       "      <td>0.238554</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>134</th>\n",
       "      <td>134</td>\n",
       "      <td>109600</td>\n",
       "      <td>1108.00</td>\n",
       "      <td>1125.00</td>\n",
       "      <td>How do you think this position happens? So mR...</td>\n",
       "      <td>[50963, 1374, 466, 345, 892, 428, 2292, 4325, ...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.154138</td>\n",
       "      <td>1.660232</td>\n",
       "      <td>0.238554</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1108.00</td>\n",
       "      <td>1110.24</td>\n",
       "      <td>How do you think this position happens?</td>\n",
       "      <td>[50363, 1374, 466, 345, 892, 428, 2292, 4325, ...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.299951</td>\n",
       "      <td>1.581395</td>\n",
       "      <td>0.019813</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>136</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1110.24</td>\n",
       "      <td>1112.96</td>\n",
       "      <td>So mRNA will have different nucleoside fields</td>\n",
       "      <td>[50475, 1406, 47227, 481, 423, 1180, 17751, 41...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.299951</td>\n",
       "      <td>1.581395</td>\n",
       "      <td>0.019813</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>137</th>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1112.96</td>\n",
       "      <td>1116.92</td>\n",
       "      <td>and I could have tRNA bringing in one amino a...</td>\n",
       "      <td>[50611, 290, 314, 714, 423, 256, 27204, 6079, ...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.299951</td>\n",
       "      <td>1.581395</td>\n",
       "      <td>0.019813</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>138</th>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>1116.92</td>\n",
       "      <td>1119.02</td>\n",
       "      <td>How do I know how does the system,</td>\n",
       "      <td>[50809, 1374, 466, 314, 760, 703, 857, 262, 10...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.299951</td>\n",
       "      <td>1.581395</td>\n",
       "      <td>0.019813</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139</th>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1119.02</td>\n",
       "      <td>1121.20</td>\n",
       "      <td>how can you encode for the fact that,</td>\n",
       "      <td>[50914, 703, 460, 345, 37773, 329, 262, 1109, ...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.299951</td>\n",
       "      <td>1.581395</td>\n",
       "      <td>0.019813</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140</th>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>1122.42</td>\n",
       "      <td>1124.52</td>\n",
       "      <td>if I hit this particular code off,</td>\n",
       "      <td>[51084, 611, 314, 2277, 428, 1948, 2438, 572, ...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.299951</td>\n",
       "      <td>1.581395</td>\n",
       "      <td>0.019813</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>141</th>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>1124.52</td>\n",
       "      <td>1126.74</td>\n",
       "      <td>I will bring in the correct amino acid.</td>\n",
       "      <td>[51189, 314, 481, 2222, 287, 262, 3376, 23206,...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.299951</td>\n",
       "      <td>1.581395</td>\n",
       "      <td>0.019813</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>142</th>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>1128.36</td>\n",
       "      <td>1129.60</td>\n",
       "      <td>There's chemistry behind it.</td>\n",
       "      <td>[51381, 1318, 338, 16585, 2157, 340, 13, 51443]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.299951</td>\n",
       "      <td>1.581395</td>\n",
       "      <td>0.019813</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>143</th>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>1129.60</td>\n",
       "      <td>1132.48</td>\n",
       "      <td>What is this position mechanism?</td>\n",
       "      <td>[51443, 1867, 318, 428, 2292, 9030, 30, 51587]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.299951</td>\n",
       "      <td>1.581395</td>\n",
       "      <td>0.019813</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>144</th>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>1132.48</td>\n",
       "      <td>1133.30</td>\n",
       "      <td>Yeah.</td>\n",
       "      <td>[51587, 9425, 13, 51628]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.299951</td>\n",
       "      <td>1.581395</td>\n",
       "      <td>0.019813</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      id    seek    start      end   \n",
       "133  133  109600  1097.00  1106.00  \\\n",
       "134  134  109600  1108.00  1125.00   \n",
       "135    0       0  1108.00  1110.24   \n",
       "136    1       0  1110.24  1112.96   \n",
       "137    2       0  1112.96  1116.92   \n",
       "138    3       0  1116.92  1119.02   \n",
       "139    4       0  1119.02  1121.20   \n",
       "140    5       0  1122.42  1124.52   \n",
       "141    6       0  1124.52  1126.74   \n",
       "142    7       0  1128.36  1129.60   \n",
       "143    8       0  1129.60  1132.48   \n",
       "144    9       0  1132.48  1133.30   \n",
       "\n",
       "                                                  text   \n",
       "133   I said coded manner because which amino acid ...  \\\n",
       "134   How do you think this position happens? So mR...   \n",
       "135            How do you think this position happens?   \n",
       "136      So mRNA will have different nucleoside fields   \n",
       "137   and I could have tRNA bringing in one amino a...   \n",
       "138                 How do I know how does the system,   \n",
       "139              how can you encode for the fact that,   \n",
       "140                 if I hit this particular code off,   \n",
       "141            I will bring in the correct amino acid.   \n",
       "142                       There's chemistry behind it.   \n",
       "143                   What is this position mechanism?   \n",
       "144                                              Yeah.   \n",
       "\n",
       "                                                tokens  temperature   \n",
       "133  [50413, 314, 531, 30817, 5642, 780, 543, 23206...          0.0  \\\n",
       "134  [50963, 1374, 466, 345, 892, 428, 2292, 4325, ...          0.0   \n",
       "135  [50363, 1374, 466, 345, 892, 428, 2292, 4325, ...          0.0   \n",
       "136  [50475, 1406, 47227, 481, 423, 1180, 17751, 41...          0.0   \n",
       "137  [50611, 290, 314, 714, 423, 256, 27204, 6079, ...          0.0   \n",
       "138  [50809, 1374, 466, 314, 760, 703, 857, 262, 10...          0.0   \n",
       "139  [50914, 703, 460, 345, 37773, 329, 262, 1109, ...          0.0   \n",
       "140  [51084, 611, 314, 2277, 428, 1948, 2438, 572, ...          0.0   \n",
       "141  [51189, 314, 481, 2222, 287, 262, 3376, 23206,...          0.0   \n",
       "142    [51381, 1318, 338, 16585, 2157, 340, 13, 51443]          0.0   \n",
       "143     [51443, 1867, 318, 428, 2292, 9030, 30, 51587]          0.0   \n",
       "144                           [51587, 9425, 13, 51628]          0.0   \n",
       "\n",
       "     avg_logprob  compression_ratio  no_speech_prob subclip_index  \n",
       "133    -0.154138           1.660232        0.238554             0  \n",
       "134    -0.154138           1.660232        0.238554             0  \n",
       "135    -0.299951           1.581395        0.019813             1  \n",
       "136    -0.299951           1.581395        0.019813             1  \n",
       "137    -0.299951           1.581395        0.019813             1  \n",
       "138    -0.299951           1.581395        0.019813             1  \n",
       "139    -0.299951           1.581395        0.019813             1  \n",
       "140    -0.299951           1.581395        0.019813             1  \n",
       "141    -0.299951           1.581395        0.019813             1  \n",
       "142    -0.299951           1.581395        0.019813             1  \n",
       "143    -0.299951           1.581395        0.019813             1  \n",
       "144    -0.299951           1.581395        0.019813             1  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subclip_0_df[133:145]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'1'"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subclip_0_df.iloc[537]['subclip_index']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>seek</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>text</th>\n",
       "      <th>tokens</th>\n",
       "      <th>temperature</th>\n",
       "      <th>avg_logprob</th>\n",
       "      <th>compression_ratio</th>\n",
       "      <th>no_speech_prob</th>\n",
       "      <th>subclip_index</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>530</th>\n",
       "      <td>395</td>\n",
       "      <td>111420</td>\n",
       "      <td>2226.72</td>\n",
       "      <td>2230.44</td>\n",
       "      <td>this five prime capping, this sort of a squar...</td>\n",
       "      <td>[50589, 428, 1936, 6994, 1275, 2105, 11, 428, ...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.220976</td>\n",
       "      <td>1.672489</td>\n",
       "      <td>0.004981</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>531</th>\n",
       "      <td>396</td>\n",
       "      <td>111420</td>\n",
       "      <td>2230.44</td>\n",
       "      <td>2232.52</td>\n",
       "      <td>and then poly-A tail tagging.</td>\n",
       "      <td>[50775, 290, 788, 7514, 12, 32, 7894, 49620, 1...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.220976</td>\n",
       "      <td>1.672489</td>\n",
       "      <td>0.004981</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>532</th>\n",
       "      <td>397</td>\n",
       "      <td>111420</td>\n",
       "      <td>2232.52</td>\n",
       "      <td>2233.72</td>\n",
       "      <td>So maybe on the footnote,</td>\n",
       "      <td>[50879, 1406, 3863, 319, 262, 38512, 11, 50939]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.220976</td>\n",
       "      <td>1.672489</td>\n",
       "      <td>0.004981</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>533</th>\n",
       "      <td>398</td>\n",
       "      <td>111420</td>\n",
       "      <td>2233.72</td>\n",
       "      <td>2236.32</td>\n",
       "      <td>I mentioned five prime cap is a purple square,</td>\n",
       "      <td>[50939, 314, 4750, 1936, 6994, 1451, 318, 257,...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.220976</td>\n",
       "      <td>1.672489</td>\n",
       "      <td>0.004981</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>534</th>\n",
       "      <td>399</td>\n",
       "      <td>111420</td>\n",
       "      <td>2236.32</td>\n",
       "      <td>2238.68</td>\n",
       "      <td>and then A is, if you see multiple A's,</td>\n",
       "      <td>[51069, 290, 788, 317, 318, 11, 611, 345, 766,...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.220976</td>\n",
       "      <td>1.672489</td>\n",
       "      <td>0.004981</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>535</th>\n",
       "      <td>400</td>\n",
       "      <td>111420</td>\n",
       "      <td>2238.68</td>\n",
       "      <td>2240.36</td>\n",
       "      <td>that's essentially poly-A tail.</td>\n",
       "      <td>[51187, 326, 338, 6986, 7514, 12, 32, 7894, 13...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.220976</td>\n",
       "      <td>1.672489</td>\n",
       "      <td>0.004981</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>536</th>\n",
       "      <td>401</td>\n",
       "      <td>111420</td>\n",
       "      <td>2240.36</td>\n",
       "      <td>2241.52</td>\n",
       "      <td>So that already tells you,</td>\n",
       "      <td>[51271, 1406, 326, 1541, 4952, 345, 11, 51329]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.220976</td>\n",
       "      <td>1.672489</td>\n",
       "      <td>0.004981</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>537</th>\n",
       "      <td>402</td>\n",
       "      <td>111420</td>\n",
       "      <td>2241.52</td>\n",
       "      <td>2245.16</td>\n",
       "      <td>even if these labels are missing,</td>\n",
       "      <td>[51329, 772, 611, 777, 14722, 389, 4814, 11, 5...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.220976</td>\n",
       "      <td>1.672489</td>\n",
       "      <td>0.004981</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>538</th>\n",
       "      <td>403</td>\n",
       "      <td>111420</td>\n",
       "      <td>2245.16</td>\n",
       "      <td>2246.32</td>\n",
       "      <td>you know that the...</td>\n",
       "      <td>[51511, 345, 760, 326, 262, 986, 51569]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.220976</td>\n",
       "      <td>1.672489</td>\n",
       "      <td>0.004981</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>539</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2226.00</td>\n",
       "      <td>2230.32</td>\n",
       "      <td>this five prime capping, this sort of a squar...</td>\n",
       "      <td>[50363, 428, 1936, 6994, 1275, 2105, 11, 428, ...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.349091</td>\n",
       "      <td>1.596639</td>\n",
       "      <td>0.241382</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>540</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>2230.32</td>\n",
       "      <td>2232.40</td>\n",
       "      <td>and then poly a tail tagging.</td>\n",
       "      <td>[50579, 290, 788, 7514, 257, 7894, 49620, 13, ...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.349091</td>\n",
       "      <td>1.596639</td>\n",
       "      <td>0.241382</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>541</th>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2232.40</td>\n",
       "      <td>2233.60</td>\n",
       "      <td>So maybe on the footnote,</td>\n",
       "      <td>[50683, 1406, 3863, 319, 262, 38512, 11, 50743]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.349091</td>\n",
       "      <td>1.596639</td>\n",
       "      <td>0.241382</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>542</th>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>2233.60</td>\n",
       "      <td>2236.20</td>\n",
       "      <td>I mentioned five prime cap is a purple square.</td>\n",
       "      <td>[50743, 314, 4750, 1936, 6994, 1451, 318, 257,...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.349091</td>\n",
       "      <td>1.596639</td>\n",
       "      <td>0.241382</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>543</th>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>2236.20</td>\n",
       "      <td>2238.56</td>\n",
       "      <td>And then a is, if you see multiple a's,</td>\n",
       "      <td>[50873, 843, 788, 257, 318, 11, 611, 345, 766,...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.349091</td>\n",
       "      <td>1.596639</td>\n",
       "      <td>0.241382</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>544</th>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>2238.56</td>\n",
       "      <td>2240.24</td>\n",
       "      <td>that's essentially poly a tail.</td>\n",
       "      <td>[50991, 326, 338, 6986, 7514, 257, 7894, 13, 5...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.349091</td>\n",
       "      <td>1.596639</td>\n",
       "      <td>0.241382</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      id    seek    start      end   \n",
       "530  395  111420  2226.72  2230.44  \\\n",
       "531  396  111420  2230.44  2232.52   \n",
       "532  397  111420  2232.52  2233.72   \n",
       "533  398  111420  2233.72  2236.32   \n",
       "534  399  111420  2236.32  2238.68   \n",
       "535  400  111420  2238.68  2240.36   \n",
       "536  401  111420  2240.36  2241.52   \n",
       "537  402  111420  2241.52  2245.16   \n",
       "538  403  111420  2245.16  2246.32   \n",
       "539    0       0  2226.00  2230.32   \n",
       "540    1       0  2230.32  2232.40   \n",
       "541    2       0  2232.40  2233.60   \n",
       "542    3       0  2233.60  2236.20   \n",
       "543    4       0  2236.20  2238.56   \n",
       "544    5       0  2238.56  2240.24   \n",
       "\n",
       "                                                  text   \n",
       "530   this five prime capping, this sort of a squar...  \\\n",
       "531                      and then poly-A tail tagging.   \n",
       "532                          So maybe on the footnote,   \n",
       "533     I mentioned five prime cap is a purple square,   \n",
       "534            and then A is, if you see multiple A's,   \n",
       "535                    that's essentially poly-A tail.   \n",
       "536                         So that already tells you,   \n",
       "537                  even if these labels are missing,   \n",
       "538                               you know that the...   \n",
       "539   this five prime capping, this sort of a squar...   \n",
       "540                      and then poly a tail tagging.   \n",
       "541                          So maybe on the footnote,   \n",
       "542     I mentioned five prime cap is a purple square.   \n",
       "543            And then a is, if you see multiple a's,   \n",
       "544                    that's essentially poly a tail.   \n",
       "\n",
       "                                                tokens  temperature   \n",
       "530  [50589, 428, 1936, 6994, 1275, 2105, 11, 428, ...          0.0  \\\n",
       "531  [50775, 290, 788, 7514, 12, 32, 7894, 49620, 1...          0.0   \n",
       "532    [50879, 1406, 3863, 319, 262, 38512, 11, 50939]          0.0   \n",
       "533  [50939, 314, 4750, 1936, 6994, 1451, 318, 257,...          0.0   \n",
       "534  [51069, 290, 788, 317, 318, 11, 611, 345, 766,...          0.0   \n",
       "535  [51187, 326, 338, 6986, 7514, 12, 32, 7894, 13...          0.0   \n",
       "536     [51271, 1406, 326, 1541, 4952, 345, 11, 51329]          0.0   \n",
       "537  [51329, 772, 611, 777, 14722, 389, 4814, 11, 5...          0.0   \n",
       "538            [51511, 345, 760, 326, 262, 986, 51569]          0.0   \n",
       "539  [50363, 428, 1936, 6994, 1275, 2105, 11, 428, ...          0.0   \n",
       "540  [50579, 290, 788, 7514, 257, 7894, 49620, 13, ...          0.0   \n",
       "541    [50683, 1406, 3863, 319, 262, 38512, 11, 50743]          0.0   \n",
       "542  [50743, 314, 4750, 1936, 6994, 1451, 318, 257,...          0.0   \n",
       "543  [50873, 843, 788, 257, 318, 11, 611, 345, 766,...          0.0   \n",
       "544  [50991, 326, 338, 6986, 7514, 257, 7894, 13, 5...          0.0   \n",
       "\n",
       "     avg_logprob  compression_ratio  no_speech_prob subclip_index  \n",
       "530    -0.220976           1.672489        0.004981             1  \n",
       "531    -0.220976           1.672489        0.004981             1  \n",
       "532    -0.220976           1.672489        0.004981             1  \n",
       "533    -0.220976           1.672489        0.004981             1  \n",
       "534    -0.220976           1.672489        0.004981             1  \n",
       "535    -0.220976           1.672489        0.004981             1  \n",
       "536    -0.220976           1.672489        0.004981             1  \n",
       "537    -0.220976           1.672489        0.004981             1  \n",
       "538    -0.220976           1.672489        0.004981             1  \n",
       "539    -0.349091           1.596639        0.241382             2  \n",
       "540    -0.349091           1.596639        0.241382             2  \n",
       "541    -0.349091           1.596639        0.241382             2  \n",
       "542    -0.349091           1.596639        0.241382             2  \n",
       "543    -0.349091           1.596639        0.241382             2  \n",
       "544    -0.349091           1.596639        0.241382             2  "
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subclip_0_df[530:545]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>seek</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>text</th>\n",
       "      <th>tokens</th>\n",
       "      <th>temperature</th>\n",
       "      <th>avg_logprob</th>\n",
       "      <th>compression_ratio</th>\n",
       "      <th>no_speech_prob</th>\n",
       "      <th>subclip_index</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>528</th>\n",
       "      <td>393</td>\n",
       "      <td>111420</td>\n",
       "      <td>2222.24</td>\n",
       "      <td>2224.44</td>\n",
       "      <td>they need to be processed and mature.</td>\n",
       "      <td>[50365, 484, 761, 284, 307, 13686, 290, 15345,...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.220976</td>\n",
       "      <td>1.672489</td>\n",
       "      <td>0.004981</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>529</th>\n",
       "      <td>394</td>\n",
       "      <td>111420</td>\n",
       "      <td>2224.44</td>\n",
       "      <td>2226.72</td>\n",
       "      <td>And this maturation process requires</td>\n",
       "      <td>[50475, 843, 428, 285, 36921, 1429, 4433, 50589]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.220976</td>\n",
       "      <td>1.672489</td>\n",
       "      <td>0.004981</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>539</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2226.00</td>\n",
       "      <td>2230.32</td>\n",
       "      <td>this five prime capping, this sort of a squar...</td>\n",
       "      <td>[50363, 428, 1936, 6994, 1275, 2105, 11, 428, ...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.349091</td>\n",
       "      <td>1.596639</td>\n",
       "      <td>0.241382</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>530</th>\n",
       "      <td>395</td>\n",
       "      <td>111420</td>\n",
       "      <td>2226.72</td>\n",
       "      <td>2230.44</td>\n",
       "      <td>this five prime capping, this sort of a squar...</td>\n",
       "      <td>[50589, 428, 1936, 6994, 1275, 2105, 11, 428, ...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.220976</td>\n",
       "      <td>1.672489</td>\n",
       "      <td>0.004981</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>540</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>2230.32</td>\n",
       "      <td>2232.40</td>\n",
       "      <td>and then poly a tail tagging.</td>\n",
       "      <td>[50579, 290, 788, 7514, 257, 7894, 49620, 13, ...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.349091</td>\n",
       "      <td>1.596639</td>\n",
       "      <td>0.241382</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>531</th>\n",
       "      <td>396</td>\n",
       "      <td>111420</td>\n",
       "      <td>2230.44</td>\n",
       "      <td>2232.52</td>\n",
       "      <td>and then poly-A tail tagging.</td>\n",
       "      <td>[50775, 290, 788, 7514, 12, 32, 7894, 49620, 1...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.220976</td>\n",
       "      <td>1.672489</td>\n",
       "      <td>0.004981</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>541</th>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2232.40</td>\n",
       "      <td>2233.60</td>\n",
       "      <td>So maybe on the footnote,</td>\n",
       "      <td>[50683, 1406, 3863, 319, 262, 38512, 11, 50743]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.349091</td>\n",
       "      <td>1.596639</td>\n",
       "      <td>0.241382</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>532</th>\n",
       "      <td>397</td>\n",
       "      <td>111420</td>\n",
       "      <td>2232.52</td>\n",
       "      <td>2233.72</td>\n",
       "      <td>So maybe on the footnote,</td>\n",
       "      <td>[50879, 1406, 3863, 319, 262, 38512, 11, 50939]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.220976</td>\n",
       "      <td>1.672489</td>\n",
       "      <td>0.004981</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>542</th>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>2233.60</td>\n",
       "      <td>2236.20</td>\n",
       "      <td>I mentioned five prime cap is a purple square.</td>\n",
       "      <td>[50743, 314, 4750, 1936, 6994, 1451, 318, 257,...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.349091</td>\n",
       "      <td>1.596639</td>\n",
       "      <td>0.241382</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>533</th>\n",
       "      <td>398</td>\n",
       "      <td>111420</td>\n",
       "      <td>2233.72</td>\n",
       "      <td>2236.32</td>\n",
       "      <td>I mentioned five prime cap is a purple square,</td>\n",
       "      <td>[50939, 314, 4750, 1936, 6994, 1451, 318, 257,...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.220976</td>\n",
       "      <td>1.672489</td>\n",
       "      <td>0.004981</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>543</th>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>2236.20</td>\n",
       "      <td>2238.56</td>\n",
       "      <td>And then a is, if you see multiple a's,</td>\n",
       "      <td>[50873, 843, 788, 257, 318, 11, 611, 345, 766,...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.349091</td>\n",
       "      <td>1.596639</td>\n",
       "      <td>0.241382</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>534</th>\n",
       "      <td>399</td>\n",
       "      <td>111420</td>\n",
       "      <td>2236.32</td>\n",
       "      <td>2238.68</td>\n",
       "      <td>and then A is, if you see multiple A's,</td>\n",
       "      <td>[51069, 290, 788, 317, 318, 11, 611, 345, 766,...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.220976</td>\n",
       "      <td>1.672489</td>\n",
       "      <td>0.004981</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>544</th>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>2238.56</td>\n",
       "      <td>2240.24</td>\n",
       "      <td>that's essentially poly a tail.</td>\n",
       "      <td>[50991, 326, 338, 6986, 7514, 257, 7894, 13, 5...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.349091</td>\n",
       "      <td>1.596639</td>\n",
       "      <td>0.241382</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>535</th>\n",
       "      <td>400</td>\n",
       "      <td>111420</td>\n",
       "      <td>2238.68</td>\n",
       "      <td>2240.36</td>\n",
       "      <td>that's essentially poly-A tail.</td>\n",
       "      <td>[51187, 326, 338, 6986, 7514, 12, 32, 7894, 13...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.220976</td>\n",
       "      <td>1.672489</td>\n",
       "      <td>0.004981</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>545</th>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>2240.24</td>\n",
       "      <td>2241.44</td>\n",
       "      <td>So that already tells you,</td>\n",
       "      <td>[51075, 1406, 326, 1541, 4952, 345, 11, 51135]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.349091</td>\n",
       "      <td>1.596639</td>\n",
       "      <td>0.241382</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>536</th>\n",
       "      <td>401</td>\n",
       "      <td>111420</td>\n",
       "      <td>2240.36</td>\n",
       "      <td>2241.52</td>\n",
       "      <td>So that already tells you,</td>\n",
       "      <td>[51271, 1406, 326, 1541, 4952, 345, 11, 51329]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.220976</td>\n",
       "      <td>1.672489</td>\n",
       "      <td>0.004981</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>546</th>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>2241.44</td>\n",
       "      <td>2245.08</td>\n",
       "      <td>even if these labels are missing,</td>\n",
       "      <td>[51135, 772, 611, 777, 14722, 389, 4814, 11, 5...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.349091</td>\n",
       "      <td>1.596639</td>\n",
       "      <td>0.241382</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      id    seek    start      end   \n",
       "528  393  111420  2222.24  2224.44  \\\n",
       "529  394  111420  2224.44  2226.72   \n",
       "539    0       0  2226.00  2230.32   \n",
       "530  395  111420  2226.72  2230.44   \n",
       "540    1       0  2230.32  2232.40   \n",
       "531  396  111420  2230.44  2232.52   \n",
       "541    2       0  2232.40  2233.60   \n",
       "532  397  111420  2232.52  2233.72   \n",
       "542    3       0  2233.60  2236.20   \n",
       "533  398  111420  2233.72  2236.32   \n",
       "543    4       0  2236.20  2238.56   \n",
       "534  399  111420  2236.32  2238.68   \n",
       "544    5       0  2238.56  2240.24   \n",
       "535  400  111420  2238.68  2240.36   \n",
       "545    6       0  2240.24  2241.44   \n",
       "536  401  111420  2240.36  2241.52   \n",
       "546    7       0  2241.44  2245.08   \n",
       "\n",
       "                                                  text   \n",
       "528              they need to be processed and mature.  \\\n",
       "529               And this maturation process requires   \n",
       "539   this five prime capping, this sort of a squar...   \n",
       "530   this five prime capping, this sort of a squar...   \n",
       "540                      and then poly a tail tagging.   \n",
       "531                      and then poly-A tail tagging.   \n",
       "541                          So maybe on the footnote,   \n",
       "532                          So maybe on the footnote,   \n",
       "542     I mentioned five prime cap is a purple square.   \n",
       "533     I mentioned five prime cap is a purple square,   \n",
       "543            And then a is, if you see multiple a's,   \n",
       "534            and then A is, if you see multiple A's,   \n",
       "544                    that's essentially poly a tail.   \n",
       "535                    that's essentially poly-A tail.   \n",
       "545                         So that already tells you,   \n",
       "536                         So that already tells you,   \n",
       "546                  even if these labels are missing,   \n",
       "\n",
       "                                                tokens  temperature   \n",
       "528  [50365, 484, 761, 284, 307, 13686, 290, 15345,...          0.0  \\\n",
       "529   [50475, 843, 428, 285, 36921, 1429, 4433, 50589]          0.0   \n",
       "539  [50363, 428, 1936, 6994, 1275, 2105, 11, 428, ...          0.0   \n",
       "530  [50589, 428, 1936, 6994, 1275, 2105, 11, 428, ...          0.0   \n",
       "540  [50579, 290, 788, 7514, 257, 7894, 49620, 13, ...          0.0   \n",
       "531  [50775, 290, 788, 7514, 12, 32, 7894, 49620, 1...          0.0   \n",
       "541    [50683, 1406, 3863, 319, 262, 38512, 11, 50743]          0.0   \n",
       "532    [50879, 1406, 3863, 319, 262, 38512, 11, 50939]          0.0   \n",
       "542  [50743, 314, 4750, 1936, 6994, 1451, 318, 257,...          0.0   \n",
       "533  [50939, 314, 4750, 1936, 6994, 1451, 318, 257,...          0.0   \n",
       "543  [50873, 843, 788, 257, 318, 11, 611, 345, 766,...          0.0   \n",
       "534  [51069, 290, 788, 317, 318, 11, 611, 345, 766,...          0.0   \n",
       "544  [50991, 326, 338, 6986, 7514, 257, 7894, 13, 5...          0.0   \n",
       "535  [51187, 326, 338, 6986, 7514, 12, 32, 7894, 13...          0.0   \n",
       "545     [51075, 1406, 326, 1541, 4952, 345, 11, 51135]          0.0   \n",
       "536     [51271, 1406, 326, 1541, 4952, 345, 11, 51329]          0.0   \n",
       "546  [51135, 772, 611, 777, 14722, 389, 4814, 11, 5...          0.0   \n",
       "\n",
       "     avg_logprob  compression_ratio  no_speech_prob subclip_index  \n",
       "528    -0.220976           1.672489        0.004981             1  \n",
       "529    -0.220976           1.672489        0.004981             1  \n",
       "539    -0.349091           1.596639        0.241382             2  \n",
       "530    -0.220976           1.672489        0.004981             1  \n",
       "540    -0.349091           1.596639        0.241382             2  \n",
       "531    -0.220976           1.672489        0.004981             1  \n",
       "541    -0.349091           1.596639        0.241382             2  \n",
       "532    -0.220976           1.672489        0.004981             1  \n",
       "542    -0.349091           1.596639        0.241382             2  \n",
       "533    -0.220976           1.672489        0.004981             1  \n",
       "543    -0.349091           1.596639        0.241382             2  \n",
       "534    -0.220976           1.672489        0.004981             1  \n",
       "544    -0.349091           1.596639        0.241382             2  \n",
       "535    -0.220976           1.672489        0.004981             1  \n",
       "545    -0.349091           1.596639        0.241382             2  \n",
       "536    -0.220976           1.672489        0.004981             1  \n",
       "546    -0.349091           1.596639        0.241382             2  "
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Show lines of the subclip_0_df dataframe with simmilar start values with one another\n",
    "\n",
    "subclip_0_df.sort_values(by=['start']).iloc[528:545]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1485\n",
      "1484\n"
     ]
    }
   ],
   "source": [
    "# Remove the rows with the same start (close to a certain tolerance) value as the previous row\n",
    "print(len(subclip_0_df))\n",
    "subclip_0_df = subclip_0_df.drop_duplicates(subset=['start'], keep='first')\n",
    "print(len(subclip_0_df))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "subclip_0_df = subclip_0_df.sort_values(by=['start'])\n",
    "subclip_0_df = subclip_0_df.reset_index(drop=True)\n",
    "\n",
    "subclip_0_df['start_diff'] = subclip_0_df['start'].diff()\n",
    "subclip_0_df['start_diff'].iloc[0] = 0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>seek</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>text</th>\n",
       "      <th>tokens</th>\n",
       "      <th>temperature</th>\n",
       "      <th>avg_logprob</th>\n",
       "      <th>compression_ratio</th>\n",
       "      <th>no_speech_prob</th>\n",
       "      <th>subclip_index</th>\n",
       "      <th>start_diff</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>528</th>\n",
       "      <td>394</td>\n",
       "      <td>111420</td>\n",
       "      <td>2224.44</td>\n",
       "      <td>2226.72</td>\n",
       "      <td>And this maturation process requires</td>\n",
       "      <td>[50475, 843, 428, 285, 36921, 1429, 4433, 50589]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.220976</td>\n",
       "      <td>1.672489</td>\n",
       "      <td>0.004981</td>\n",
       "      <td>1</td>\n",
       "      <td>2.20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>529</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2226.00</td>\n",
       "      <td>2230.32</td>\n",
       "      <td>this five prime capping, this sort of a squar...</td>\n",
       "      <td>[50363, 428, 1936, 6994, 1275, 2105, 11, 428, ...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.349091</td>\n",
       "      <td>1.596639</td>\n",
       "      <td>0.241382</td>\n",
       "      <td>2</td>\n",
       "      <td>1.56</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>530</th>\n",
       "      <td>395</td>\n",
       "      <td>111420</td>\n",
       "      <td>2226.72</td>\n",
       "      <td>2230.44</td>\n",
       "      <td>this five prime capping, this sort of a squar...</td>\n",
       "      <td>[50589, 428, 1936, 6994, 1275, 2105, 11, 428, ...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.220976</td>\n",
       "      <td>1.672489</td>\n",
       "      <td>0.004981</td>\n",
       "      <td>1</td>\n",
       "      <td>0.72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>531</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>2230.32</td>\n",
       "      <td>2232.40</td>\n",
       "      <td>and then poly a tail tagging.</td>\n",
       "      <td>[50579, 290, 788, 7514, 257, 7894, 49620, 13, ...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.349091</td>\n",
       "      <td>1.596639</td>\n",
       "      <td>0.241382</td>\n",
       "      <td>2</td>\n",
       "      <td>3.60</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>532</th>\n",
       "      <td>396</td>\n",
       "      <td>111420</td>\n",
       "      <td>2230.44</td>\n",
       "      <td>2232.52</td>\n",
       "      <td>and then poly-A tail tagging.</td>\n",
       "      <td>[50775, 290, 788, 7514, 12, 32, 7894, 49620, 1...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.220976</td>\n",
       "      <td>1.672489</td>\n",
       "      <td>0.004981</td>\n",
       "      <td>1</td>\n",
       "      <td>0.12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>533</th>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2232.40</td>\n",
       "      <td>2233.60</td>\n",
       "      <td>So maybe on the footnote,</td>\n",
       "      <td>[50683, 1406, 3863, 319, 262, 38512, 11, 50743]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.349091</td>\n",
       "      <td>1.596639</td>\n",
       "      <td>0.241382</td>\n",
       "      <td>2</td>\n",
       "      <td>1.96</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>534</th>\n",
       "      <td>397</td>\n",
       "      <td>111420</td>\n",
       "      <td>2232.52</td>\n",
       "      <td>2233.72</td>\n",
       "      <td>So maybe on the footnote,</td>\n",
       "      <td>[50879, 1406, 3863, 319, 262, 38512, 11, 50939]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.220976</td>\n",
       "      <td>1.672489</td>\n",
       "      <td>0.004981</td>\n",
       "      <td>1</td>\n",
       "      <td>0.12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>535</th>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>2233.60</td>\n",
       "      <td>2236.20</td>\n",
       "      <td>I mentioned five prime cap is a purple square.</td>\n",
       "      <td>[50743, 314, 4750, 1936, 6994, 1451, 318, 257,...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.349091</td>\n",
       "      <td>1.596639</td>\n",
       "      <td>0.241382</td>\n",
       "      <td>2</td>\n",
       "      <td>1.08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>536</th>\n",
       "      <td>398</td>\n",
       "      <td>111420</td>\n",
       "      <td>2233.72</td>\n",
       "      <td>2236.32</td>\n",
       "      <td>I mentioned five prime cap is a purple square,</td>\n",
       "      <td>[50939, 314, 4750, 1936, 6994, 1451, 318, 257,...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.220976</td>\n",
       "      <td>1.672489</td>\n",
       "      <td>0.004981</td>\n",
       "      <td>1</td>\n",
       "      <td>0.12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>537</th>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>2236.20</td>\n",
       "      <td>2238.56</td>\n",
       "      <td>And then a is, if you see multiple a's,</td>\n",
       "      <td>[50873, 843, 788, 257, 318, 11, 611, 345, 766,...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.349091</td>\n",
       "      <td>1.596639</td>\n",
       "      <td>0.241382</td>\n",
       "      <td>2</td>\n",
       "      <td>2.48</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>538</th>\n",
       "      <td>399</td>\n",
       "      <td>111420</td>\n",
       "      <td>2236.32</td>\n",
       "      <td>2238.68</td>\n",
       "      <td>and then A is, if you see multiple A's,</td>\n",
       "      <td>[51069, 290, 788, 317, 318, 11, 611, 345, 766,...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.220976</td>\n",
       "      <td>1.672489</td>\n",
       "      <td>0.004981</td>\n",
       "      <td>1</td>\n",
       "      <td>0.12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>539</th>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>2238.56</td>\n",
       "      <td>2240.24</td>\n",
       "      <td>that's essentially poly a tail.</td>\n",
       "      <td>[50991, 326, 338, 6986, 7514, 257, 7894, 13, 5...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.349091</td>\n",
       "      <td>1.596639</td>\n",
       "      <td>0.241382</td>\n",
       "      <td>2</td>\n",
       "      <td>2.24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>540</th>\n",
       "      <td>400</td>\n",
       "      <td>111420</td>\n",
       "      <td>2238.68</td>\n",
       "      <td>2240.36</td>\n",
       "      <td>that's essentially poly-A tail.</td>\n",
       "      <td>[51187, 326, 338, 6986, 7514, 12, 32, 7894, 13...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.220976</td>\n",
       "      <td>1.672489</td>\n",
       "      <td>0.004981</td>\n",
       "      <td>1</td>\n",
       "      <td>0.12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>541</th>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>2240.24</td>\n",
       "      <td>2241.44</td>\n",
       "      <td>So that already tells you,</td>\n",
       "      <td>[51075, 1406, 326, 1541, 4952, 345, 11, 51135]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.349091</td>\n",
       "      <td>1.596639</td>\n",
       "      <td>0.241382</td>\n",
       "      <td>2</td>\n",
       "      <td>1.56</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>542</th>\n",
       "      <td>401</td>\n",
       "      <td>111420</td>\n",
       "      <td>2240.36</td>\n",
       "      <td>2241.52</td>\n",
       "      <td>So that already tells you,</td>\n",
       "      <td>[51271, 1406, 326, 1541, 4952, 345, 11, 51329]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.220976</td>\n",
       "      <td>1.672489</td>\n",
       "      <td>0.004981</td>\n",
       "      <td>1</td>\n",
       "      <td>0.12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>543</th>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>2241.44</td>\n",
       "      <td>2245.08</td>\n",
       "      <td>even if these labels are missing,</td>\n",
       "      <td>[51135, 772, 611, 777, 14722, 389, 4814, 11, 5...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.349091</td>\n",
       "      <td>1.596639</td>\n",
       "      <td>0.241382</td>\n",
       "      <td>2</td>\n",
       "      <td>1.08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>544</th>\n",
       "      <td>402</td>\n",
       "      <td>111420</td>\n",
       "      <td>2241.52</td>\n",
       "      <td>2245.16</td>\n",
       "      <td>even if these labels are missing,</td>\n",
       "      <td>[51329, 772, 611, 777, 14722, 389, 4814, 11, 5...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.220976</td>\n",
       "      <td>1.672489</td>\n",
       "      <td>0.004981</td>\n",
       "      <td>1</td>\n",
       "      <td>0.08</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      id    seek    start      end   \n",
       "528  394  111420  2224.44  2226.72  \\\n",
       "529    0       0  2226.00  2230.32   \n",
       "530  395  111420  2226.72  2230.44   \n",
       "531    1       0  2230.32  2232.40   \n",
       "532  396  111420  2230.44  2232.52   \n",
       "533    2       0  2232.40  2233.60   \n",
       "534  397  111420  2232.52  2233.72   \n",
       "535    3       0  2233.60  2236.20   \n",
       "536  398  111420  2233.72  2236.32   \n",
       "537    4       0  2236.20  2238.56   \n",
       "538  399  111420  2236.32  2238.68   \n",
       "539    5       0  2238.56  2240.24   \n",
       "540  400  111420  2238.68  2240.36   \n",
       "541    6       0  2240.24  2241.44   \n",
       "542  401  111420  2240.36  2241.52   \n",
       "543    7       0  2241.44  2245.08   \n",
       "544  402  111420  2241.52  2245.16   \n",
       "\n",
       "                                                  text   \n",
       "528               And this maturation process requires  \\\n",
       "529   this five prime capping, this sort of a squar...   \n",
       "530   this five prime capping, this sort of a squar...   \n",
       "531                      and then poly a tail tagging.   \n",
       "532                      and then poly-A tail tagging.   \n",
       "533                          So maybe on the footnote,   \n",
       "534                          So maybe on the footnote,   \n",
       "535     I mentioned five prime cap is a purple square.   \n",
       "536     I mentioned five prime cap is a purple square,   \n",
       "537            And then a is, if you see multiple a's,   \n",
       "538            and then A is, if you see multiple A's,   \n",
       "539                    that's essentially poly a tail.   \n",
       "540                    that's essentially poly-A tail.   \n",
       "541                         So that already tells you,   \n",
       "542                         So that already tells you,   \n",
       "543                  even if these labels are missing,   \n",
       "544                  even if these labels are missing,   \n",
       "\n",
       "                                                tokens  temperature   \n",
       "528   [50475, 843, 428, 285, 36921, 1429, 4433, 50589]          0.0  \\\n",
       "529  [50363, 428, 1936, 6994, 1275, 2105, 11, 428, ...          0.0   \n",
       "530  [50589, 428, 1936, 6994, 1275, 2105, 11, 428, ...          0.0   \n",
       "531  [50579, 290, 788, 7514, 257, 7894, 49620, 13, ...          0.0   \n",
       "532  [50775, 290, 788, 7514, 12, 32, 7894, 49620, 1...          0.0   \n",
       "533    [50683, 1406, 3863, 319, 262, 38512, 11, 50743]          0.0   \n",
       "534    [50879, 1406, 3863, 319, 262, 38512, 11, 50939]          0.0   \n",
       "535  [50743, 314, 4750, 1936, 6994, 1451, 318, 257,...          0.0   \n",
       "536  [50939, 314, 4750, 1936, 6994, 1451, 318, 257,...          0.0   \n",
       "537  [50873, 843, 788, 257, 318, 11, 611, 345, 766,...          0.0   \n",
       "538  [51069, 290, 788, 317, 318, 11, 611, 345, 766,...          0.0   \n",
       "539  [50991, 326, 338, 6986, 7514, 257, 7894, 13, 5...          0.0   \n",
       "540  [51187, 326, 338, 6986, 7514, 12, 32, 7894, 13...          0.0   \n",
       "541     [51075, 1406, 326, 1541, 4952, 345, 11, 51135]          0.0   \n",
       "542     [51271, 1406, 326, 1541, 4952, 345, 11, 51329]          0.0   \n",
       "543  [51135, 772, 611, 777, 14722, 389, 4814, 11, 5...          0.0   \n",
       "544  [51329, 772, 611, 777, 14722, 389, 4814, 11, 5...          0.0   \n",
       "\n",
       "     avg_logprob  compression_ratio  no_speech_prob subclip_index  start_diff  \n",
       "528    -0.220976           1.672489        0.004981             1        2.20  \n",
       "529    -0.349091           1.596639        0.241382             2        1.56  \n",
       "530    -0.220976           1.672489        0.004981             1        0.72  \n",
       "531    -0.349091           1.596639        0.241382             2        3.60  \n",
       "532    -0.220976           1.672489        0.004981             1        0.12  \n",
       "533    -0.349091           1.596639        0.241382             2        1.96  \n",
       "534    -0.220976           1.672489        0.004981             1        0.12  \n",
       "535    -0.349091           1.596639        0.241382             2        1.08  \n",
       "536    -0.220976           1.672489        0.004981             1        0.12  \n",
       "537    -0.349091           1.596639        0.241382             2        2.48  \n",
       "538    -0.220976           1.672489        0.004981             1        0.12  \n",
       "539    -0.349091           1.596639        0.241382             2        2.24  \n",
       "540    -0.220976           1.672489        0.004981             1        0.12  \n",
       "541    -0.349091           1.596639        0.241382             2        1.56  \n",
       "542    -0.220976           1.672489        0.004981             1        0.12  \n",
       "543    -0.349091           1.596639        0.241382             2        1.08  \n",
       "544    -0.220976           1.672489        0.004981             1        0.08  "
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subclip_0_df.iloc[528:545]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>seek</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>text</th>\n",
       "      <th>tokens</th>\n",
       "      <th>temperature</th>\n",
       "      <th>avg_logprob</th>\n",
       "      <th>compression_ratio</th>\n",
       "      <th>no_speech_prob</th>\n",
       "      <th>subclip_index</th>\n",
       "      <th>start_diff</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>532</th>\n",
       "      <td>396</td>\n",
       "      <td>111420</td>\n",
       "      <td>2230.44</td>\n",
       "      <td>2232.52</td>\n",
       "      <td>and then poly-A tail tagging.</td>\n",
       "      <td>[50775, 290, 788, 7514, 12, 32, 7894, 49620, 1...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.220976</td>\n",
       "      <td>1.672489</td>\n",
       "      <td>0.004981</td>\n",
       "      <td>1</td>\n",
       "      <td>0.12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>534</th>\n",
       "      <td>397</td>\n",
       "      <td>111420</td>\n",
       "      <td>2232.52</td>\n",
       "      <td>2233.72</td>\n",
       "      <td>So maybe on the footnote,</td>\n",
       "      <td>[50879, 1406, 3863, 319, 262, 38512, 11, 50939]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.220976</td>\n",
       "      <td>1.672489</td>\n",
       "      <td>0.004981</td>\n",
       "      <td>1</td>\n",
       "      <td>0.12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>536</th>\n",
       "      <td>398</td>\n",
       "      <td>111420</td>\n",
       "      <td>2233.72</td>\n",
       "      <td>2236.32</td>\n",
       "      <td>I mentioned five prime cap is a purple square,</td>\n",
       "      <td>[50939, 314, 4750, 1936, 6994, 1451, 318, 257,...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.220976</td>\n",
       "      <td>1.672489</td>\n",
       "      <td>0.004981</td>\n",
       "      <td>1</td>\n",
       "      <td>0.12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>538</th>\n",
       "      <td>399</td>\n",
       "      <td>111420</td>\n",
       "      <td>2236.32</td>\n",
       "      <td>2238.68</td>\n",
       "      <td>and then A is, if you see multiple A's,</td>\n",
       "      <td>[51069, 290, 788, 317, 318, 11, 611, 345, 766,...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.220976</td>\n",
       "      <td>1.672489</td>\n",
       "      <td>0.004981</td>\n",
       "      <td>1</td>\n",
       "      <td>0.12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>540</th>\n",
       "      <td>400</td>\n",
       "      <td>111420</td>\n",
       "      <td>2238.68</td>\n",
       "      <td>2240.36</td>\n",
       "      <td>that's essentially poly-A tail.</td>\n",
       "      <td>[51187, 326, 338, 6986, 7514, 12, 32, 7894, 13...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.220976</td>\n",
       "      <td>1.672489</td>\n",
       "      <td>0.004981</td>\n",
       "      <td>1</td>\n",
       "      <td>0.12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>542</th>\n",
       "      <td>401</td>\n",
       "      <td>111420</td>\n",
       "      <td>2240.36</td>\n",
       "      <td>2241.52</td>\n",
       "      <td>So that already tells you,</td>\n",
       "      <td>[51271, 1406, 326, 1541, 4952, 345, 11, 51329]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.220976</td>\n",
       "      <td>1.672489</td>\n",
       "      <td>0.004981</td>\n",
       "      <td>1</td>\n",
       "      <td>0.12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>544</th>\n",
       "      <td>402</td>\n",
       "      <td>111420</td>\n",
       "      <td>2241.52</td>\n",
       "      <td>2245.16</td>\n",
       "      <td>even if these labels are missing,</td>\n",
       "      <td>[51329, 772, 611, 777, 14722, 389, 4814, 11, 5...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.220976</td>\n",
       "      <td>1.672489</td>\n",
       "      <td>0.004981</td>\n",
       "      <td>1</td>\n",
       "      <td>0.08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>546</th>\n",
       "      <td>403</td>\n",
       "      <td>111420</td>\n",
       "      <td>2245.16</td>\n",
       "      <td>2246.32</td>\n",
       "      <td>you know that the...</td>\n",
       "      <td>[51511, 345, 760, 326, 262, 986, 51569]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.220976</td>\n",
       "      <td>1.672489</td>\n",
       "      <td>0.004981</td>\n",
       "      <td>1</td>\n",
       "      <td>0.08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>927</th>\n",
       "      <td>387</td>\n",
       "      <td>111846</td>\n",
       "      <td>3353.06</td>\n",
       "      <td>3356.38</td>\n",
       "      <td>where if the gene transcription is successful,</td>\n",
       "      <td>[50793, 810, 611, 262, 9779, 26955, 318, 4388,...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.317962</td>\n",
       "      <td>1.436782</td>\n",
       "      <td>0.010407</td>\n",
       "      <td>2</td>\n",
       "      <td>0.02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>929</th>\n",
       "      <td>388</td>\n",
       "      <td>111846</td>\n",
       "      <td>3356.38</td>\n",
       "      <td>3358.94</td>\n",
       "      <td>you will either express this either His3 gene</td>\n",
       "      <td>[50959, 345, 481, 2035, 4911, 428, 2035, 2399,...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.317962</td>\n",
       "      <td>1.436782</td>\n",
       "      <td>0.010407</td>\n",
       "      <td>2</td>\n",
       "      <td>0.04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>931</th>\n",
       "      <td>389</td>\n",
       "      <td>111846</td>\n",
       "      <td>3358.94</td>\n",
       "      <td>3360.14</td>\n",
       "      <td>or the Lexi gene.</td>\n",
       "      <td>[51087, 393, 262, 17210, 72, 9779, 13, 51147]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.317962</td>\n",
       "      <td>1.436782</td>\n",
       "      <td>0.010407</td>\n",
       "      <td>2</td>\n",
       "      <td>0.04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>933</th>\n",
       "      <td>390</td>\n",
       "      <td>111846</td>\n",
       "      <td>3360.14</td>\n",
       "      <td>3362.02</td>\n",
       "      <td>This is just a, you know, example.</td>\n",
       "      <td>[51147, 770, 318, 655, 257, 11, 345, 760, 11, ...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.317962</td>\n",
       "      <td>1.436782</td>\n",
       "      <td>0.010407</td>\n",
       "      <td>2</td>\n",
       "      <td>0.06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>935</th>\n",
       "      <td>391</td>\n",
       "      <td>111846</td>\n",
       "      <td>3362.02</td>\n",
       "      <td>3364.14</td>\n",
       "      <td>It's not fixed to that system.</td>\n",
       "      <td>[51241, 632, 338, 407, 5969, 284, 326, 1080, 1...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.317962</td>\n",
       "      <td>1.436782</td>\n",
       "      <td>0.010407</td>\n",
       "      <td>2</td>\n",
       "      <td>0.06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1312</th>\n",
       "      <td>380</td>\n",
       "      <td>109516</td>\n",
       "      <td>4466.92</td>\n",
       "      <td>4468.84</td>\n",
       "      <td>So this will be completely off.</td>\n",
       "      <td>[51751, 1406, 428, 481, 307, 3190, 572, 13, 51...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.501886</td>\n",
       "      <td>1.750988</td>\n",
       "      <td>0.012048</td>\n",
       "      <td>3</td>\n",
       "      <td>0.12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1316</th>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>4471.12</td>\n",
       "      <td>4476.96</td>\n",
       "      <td>is completely non-controlled. So you'll have ...</td>\n",
       "      <td>[50819, 318, 3190, 1729, 12, 14401, 13, 1406, ...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.195802</td>\n",
       "      <td>1.740458</td>\n",
       "      <td>0.057309</td>\n",
       "      <td>4</td>\n",
       "      <td>0.04</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       id    seek    start      end   \n",
       "532   396  111420  2230.44  2232.52  \\\n",
       "534   397  111420  2232.52  2233.72   \n",
       "536   398  111420  2233.72  2236.32   \n",
       "538   399  111420  2236.32  2238.68   \n",
       "540   400  111420  2238.68  2240.36   \n",
       "542   401  111420  2240.36  2241.52   \n",
       "544   402  111420  2241.52  2245.16   \n",
       "546   403  111420  2245.16  2246.32   \n",
       "927   387  111846  3353.06  3356.38   \n",
       "929   388  111846  3356.38  3358.94   \n",
       "931   389  111846  3358.94  3360.14   \n",
       "933   390  111846  3360.14  3362.02   \n",
       "935   391  111846  3362.02  3364.14   \n",
       "1312  380  109516  4466.92  4468.84   \n",
       "1316    2       0  4471.12  4476.96   \n",
       "\n",
       "                                                   text   \n",
       "532                       and then poly-A tail tagging.  \\\n",
       "534                           So maybe on the footnote,   \n",
       "536      I mentioned five prime cap is a purple square,   \n",
       "538             and then A is, if you see multiple A's,   \n",
       "540                     that's essentially poly-A tail.   \n",
       "542                          So that already tells you,   \n",
       "544                   even if these labels are missing,   \n",
       "546                                you know that the...   \n",
       "927      where if the gene transcription is successful,   \n",
       "929       you will either express this either His3 gene   \n",
       "931                                   or the Lexi gene.   \n",
       "933                  This is just a, you know, example.   \n",
       "935                      It's not fixed to that system.   \n",
       "1312                    So this will be completely off.   \n",
       "1316   is completely non-controlled. So you'll have ...   \n",
       "\n",
       "                                                 tokens  temperature   \n",
       "532   [50775, 290, 788, 7514, 12, 32, 7894, 49620, 1...          0.0  \\\n",
       "534     [50879, 1406, 3863, 319, 262, 38512, 11, 50939]          0.0   \n",
       "536   [50939, 314, 4750, 1936, 6994, 1451, 318, 257,...          0.0   \n",
       "538   [51069, 290, 788, 317, 318, 11, 611, 345, 766,...          0.0   \n",
       "540   [51187, 326, 338, 6986, 7514, 12, 32, 7894, 13...          0.0   \n",
       "542      [51271, 1406, 326, 1541, 4952, 345, 11, 51329]          0.0   \n",
       "544   [51329, 772, 611, 777, 14722, 389, 4814, 11, 5...          0.0   \n",
       "546             [51511, 345, 760, 326, 262, 986, 51569]          0.0   \n",
       "927   [50793, 810, 611, 262, 9779, 26955, 318, 4388,...          0.0   \n",
       "929   [50959, 345, 481, 2035, 4911, 428, 2035, 2399,...          0.0   \n",
       "931       [51087, 393, 262, 17210, 72, 9779, 13, 51147]          0.0   \n",
       "933   [51147, 770, 318, 655, 257, 11, 345, 760, 11, ...          0.0   \n",
       "935   [51241, 632, 338, 407, 5969, 284, 326, 1080, 1...          0.0   \n",
       "1312  [51751, 1406, 428, 481, 307, 3190, 572, 13, 51...          0.0   \n",
       "1316  [50819, 318, 3190, 1729, 12, 14401, 13, 1406, ...          0.0   \n",
       "\n",
       "      avg_logprob  compression_ratio  no_speech_prob subclip_index  start_diff  \n",
       "532     -0.220976           1.672489        0.004981             1        0.12  \n",
       "534     -0.220976           1.672489        0.004981             1        0.12  \n",
       "536     -0.220976           1.672489        0.004981             1        0.12  \n",
       "538     -0.220976           1.672489        0.004981             1        0.12  \n",
       "540     -0.220976           1.672489        0.004981             1        0.12  \n",
       "542     -0.220976           1.672489        0.004981             1        0.12  \n",
       "544     -0.220976           1.672489        0.004981             1        0.08  \n",
       "546     -0.220976           1.672489        0.004981             1        0.08  \n",
       "927     -0.317962           1.436782        0.010407             2        0.02  \n",
       "929     -0.317962           1.436782        0.010407             2        0.04  \n",
       "931     -0.317962           1.436782        0.010407             2        0.04  \n",
       "933     -0.317962           1.436782        0.010407             2        0.06  \n",
       "935     -0.317962           1.436782        0.010407             2        0.06  \n",
       "1312    -0.501886           1.750988        0.012048             3        0.12  \n",
       "1316    -0.195802           1.740458        0.057309             4        0.04  "
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subclip_0_df.sort_values(by=['start']).head(10)\n",
    "\n",
    "# Show lines where start_diff is smaller than 0.5\n",
    "subclip_0_df[subclip_0_df['start_diff'] < 0.5]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>seek</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>text</th>\n",
       "      <th>tokens</th>\n",
       "      <th>temperature</th>\n",
       "      <th>avg_logprob</th>\n",
       "      <th>compression_ratio</th>\n",
       "      <th>no_speech_prob</th>\n",
       "      <th>subclip_index</th>\n",
       "      <th>start_diff</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1308</th>\n",
       "      <td>378</td>\n",
       "      <td>109516</td>\n",
       "      <td>4460.36</td>\n",
       "      <td>4463.96</td>\n",
       "      <td>No, so essentially if you essentially have th...</td>\n",
       "      <td>[51423, 1400, 11, 523, 6986, 611, 345, 6986, 4...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.501886</td>\n",
       "      <td>1.750988</td>\n",
       "      <td>0.012048</td>\n",
       "      <td>3</td>\n",
       "      <td>3.12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1309</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4462.00</td>\n",
       "      <td>4466.80</td>\n",
       "      <td>essentially have this sequence, there will no...</td>\n",
       "      <td>[50363, 6986, 423, 428, 8379, 11, 612, 481, 40...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.195802</td>\n",
       "      <td>1.740458</td>\n",
       "      <td>0.057309</td>\n",
       "      <td>4</td>\n",
       "      <td>1.64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1310</th>\n",
       "      <td>379</td>\n",
       "      <td>109516</td>\n",
       "      <td>4463.96</td>\n",
       "      <td>4466.92</td>\n",
       "      <td>there will not be any gene expression in the ...</td>\n",
       "      <td>[51603, 612, 481, 407, 307, 597, 9779, 5408, 2...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.501886</td>\n",
       "      <td>1.750988</td>\n",
       "      <td>0.012048</td>\n",
       "      <td>3</td>\n",
       "      <td>1.96</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1311</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>4466.80</td>\n",
       "      <td>4471.12</td>\n",
       "      <td>So this will be completely off. This may be c...</td>\n",
       "      <td>[50603, 1406, 428, 481, 307, 3190, 572, 13, 77...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.195802</td>\n",
       "      <td>1.740458</td>\n",
       "      <td>0.057309</td>\n",
       "      <td>4</td>\n",
       "      <td>2.84</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1312</th>\n",
       "      <td>380</td>\n",
       "      <td>109516</td>\n",
       "      <td>4466.92</td>\n",
       "      <td>4468.84</td>\n",
       "      <td>So this will be completely off.</td>\n",
       "      <td>[51751, 1406, 428, 481, 307, 3190, 572, 13, 51...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.501886</td>\n",
       "      <td>1.750988</td>\n",
       "      <td>0.012048</td>\n",
       "      <td>3</td>\n",
       "      <td>0.12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1313</th>\n",
       "      <td>381</td>\n",
       "      <td>112484</td>\n",
       "      <td>4469.40</td>\n",
       "      <td>4470.24</td>\n",
       "      <td>This may be confusing.</td>\n",
       "      <td>[50391, 770, 743, 307, 15337, 13, 50433]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.211209</td>\n",
       "      <td>1.389262</td>\n",
       "      <td>0.027582</td>\n",
       "      <td>3</td>\n",
       "      <td>2.48</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1314</th>\n",
       "      <td>382</td>\n",
       "      <td>112484</td>\n",
       "      <td>4470.24</td>\n",
       "      <td>4471.08</td>\n",
       "      <td>This is essentially the system</td>\n",
       "      <td>[50433, 770, 318, 6986, 262, 1080, 50475]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.211209</td>\n",
       "      <td>1.389262</td>\n",
       "      <td>0.027582</td>\n",
       "      <td>3</td>\n",
       "      <td>0.84</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1315</th>\n",
       "      <td>383</td>\n",
       "      <td>112484</td>\n",
       "      <td>4471.08</td>\n",
       "      <td>4473.56</td>\n",
       "      <td>that is completely non-controlled.</td>\n",
       "      <td>[50475, 326, 318, 3190, 1729, 12, 14401, 13, 5...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.211209</td>\n",
       "      <td>1.389262</td>\n",
       "      <td>0.027582</td>\n",
       "      <td>3</td>\n",
       "      <td>0.84</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1316</th>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>4471.12</td>\n",
       "      <td>4476.96</td>\n",
       "      <td>is completely non-controlled. So you'll have ...</td>\n",
       "      <td>[50819, 318, 3190, 1729, 12, 14401, 13, 1406, ...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.195802</td>\n",
       "      <td>1.740458</td>\n",
       "      <td>0.057309</td>\n",
       "      <td>4</td>\n",
       "      <td>0.04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1317</th>\n",
       "      <td>384</td>\n",
       "      <td>112484</td>\n",
       "      <td>4473.56</td>\n",
       "      <td>4476.16</td>\n",
       "      <td>So you'll have a GFP expression as is shown</td>\n",
       "      <td>[50599, 1406, 345, 1183, 423, 257, 402, 5837, ...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.211209</td>\n",
       "      <td>1.389262</td>\n",
       "      <td>0.027582</td>\n",
       "      <td>3</td>\n",
       "      <td>2.44</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       id    seek    start      end   \n",
       "1308  378  109516  4460.36  4463.96  \\\n",
       "1309    0       0  4462.00  4466.80   \n",
       "1310  379  109516  4463.96  4466.92   \n",
       "1311    1       0  4466.80  4471.12   \n",
       "1312  380  109516  4466.92  4468.84   \n",
       "1313  381  112484  4469.40  4470.24   \n",
       "1314  382  112484  4470.24  4471.08   \n",
       "1315  383  112484  4471.08  4473.56   \n",
       "1316    2       0  4471.12  4476.96   \n",
       "1317  384  112484  4473.56  4476.16   \n",
       "\n",
       "                                                   text   \n",
       "1308   No, so essentially if you essentially have th...  \\\n",
       "1309   essentially have this sequence, there will no...   \n",
       "1310   there will not be any gene expression in the ...   \n",
       "1311   So this will be completely off. This may be c...   \n",
       "1312                    So this will be completely off.   \n",
       "1313                             This may be confusing.   \n",
       "1314                     This is essentially the system   \n",
       "1315                 that is completely non-controlled.   \n",
       "1316   is completely non-controlled. So you'll have ...   \n",
       "1317        So you'll have a GFP expression as is shown   \n",
       "\n",
       "                                                 tokens  temperature   \n",
       "1308  [51423, 1400, 11, 523, 6986, 611, 345, 6986, 4...          0.0  \\\n",
       "1309  [50363, 6986, 423, 428, 8379, 11, 612, 481, 40...          0.0   \n",
       "1310  [51603, 612, 481, 407, 307, 597, 9779, 5408, 2...          0.0   \n",
       "1311  [50603, 1406, 428, 481, 307, 3190, 572, 13, 77...          0.0   \n",
       "1312  [51751, 1406, 428, 481, 307, 3190, 572, 13, 51...          0.0   \n",
       "1313           [50391, 770, 743, 307, 15337, 13, 50433]          0.0   \n",
       "1314          [50433, 770, 318, 6986, 262, 1080, 50475]          0.0   \n",
       "1315  [50475, 326, 318, 3190, 1729, 12, 14401, 13, 5...          0.0   \n",
       "1316  [50819, 318, 3190, 1729, 12, 14401, 13, 1406, ...          0.0   \n",
       "1317  [50599, 1406, 345, 1183, 423, 257, 402, 5837, ...          0.0   \n",
       "\n",
       "      avg_logprob  compression_ratio  no_speech_prob subclip_index  start_diff  \n",
       "1308    -0.501886           1.750988        0.012048             3        3.12  \n",
       "1309    -0.195802           1.740458        0.057309             4        1.64  \n",
       "1310    -0.501886           1.750988        0.012048             3        1.96  \n",
       "1311    -0.195802           1.740458        0.057309             4        2.84  \n",
       "1312    -0.501886           1.750988        0.012048             3        0.12  \n",
       "1313    -0.211209           1.389262        0.027582             3        2.48  \n",
       "1314    -0.211209           1.389262        0.027582             3        0.84  \n",
       "1315    -0.211209           1.389262        0.027582             3        0.84  \n",
       "1316    -0.195802           1.740458        0.057309             4        0.04  \n",
       "1317    -0.211209           1.389262        0.027582             3        2.44  "
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subclip_0_df[1308:1318]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "speaker-notes",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.16"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
